# Resources

<!-- --- -->

<!-- Please find a link to today's presentation slides [here](../../enrichment_analysis_2024.pdf) -->

<!-- ----- -->

The following tools might be useful for downstream functional analysis; this includes some not covered in today's workshop. 

<!-- https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/shsy5ydiff.png) 
http://localhost:1313/enrichment_analysis_workshop/img/david.png
--> 

Note that most of these tools do more than just enrichment tests, and some include their own databases. 

## g:Profiler / g:GOSt

https://biit.cs.ut.ee/gprofiler/gost

The `g:GOSt` "functional profiling" tool of `gProfiler` calculates functional enrichment for gene lists, identifying enriched terms across various ontologies and pathways. 
It features a clean, modern interface and provides a handy summary highlighting which genes contribute to the enrichment.

<!-- ![](https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/gprofiler.png) -->
```{r, echo=FALSE, out.width="100%", fig.align = "centre", fig.cap="The gProfiler front page"} 
knitr::include_graphics("https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/gprofiler.png")
```


## PANTHER 

http://www.pantherdb.org/

PANTHER performs over-representation tests across multiple databases; Gene ontology, Reactome, PANTHER pathways and protein classes. It allows greater control over the statistical test used and clearly summarises what was performed.

<!-- ![](https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/panther.png) -->
```{r, echo=FALSE, out.width="100%", fig.align = "centre", fig.cap="PANTHER"} 
knitr::include_graphics("https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/panther.png")
```


## DAVID 

https://david.ncifcrf.gov/

Via its 'functional annotation' tool, DAVID allows you to calculate functional enrichment across a number of databases, including Gene Ontology, KEGG, Reactome, and others. Reliable, with a slightly clunky interface.

<!-- ![](https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/david.png) -->
```{r, echo=FALSE, out.width="100%", fig.align = "centre", fig.cap="DAVID"} 
knitr::include_graphics("https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/david.png")
```


## Enrichr

https://amp.pharm.mssm.edu/Enrichr/

Enrichr easily calculates enrichment across a wide range of databases. It currently allows for a background set. 

<!-- ![](https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/enrichr.png) -->
```{r, echo=FALSE, out.width="100%", fig.align = "centre", fig.cap="Enrichr"} 
knitr::include_graphics("https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/enrichr.png")
```



## Reactome

https://reactome.org/

The core of Reactome is the Reactome Pathway Browser. Although other tools use the Reactome database, the Reactome website provides a means to browse enrichment within the pathway browser view.

<!-- ![](https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/reactome.png) -->
```{r, echo=FALSE, out.width="100%", fig.align = "centre", fig.cap="Reactome"} 
knitr::include_graphics("https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/reactome.png")
```


## BioCyc

https://biocyc.org/

BioCyc is another suite of tools for enrichment and pathway browsing, particularly useful for prokaryotic studies. It is a licensed resource, but Monash holds an institutional licence.

<!-- ![](https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/biocyc.png) -->
```{r, echo=FALSE, out.width="100%", fig.align = "centre", fig.cap="Biocyc"} 
knitr::include_graphics("https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/biocyc.png")
```


## STRING

https://string-db.org/

STRING was originally designed as a convenient tool to explore interactions within a gene list, visualised as an interaction network. While it is best suited for smaller gene lists, it now also includes functionality for performing functional enrichment analysis.

<!-- ![](https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/string.png) -->
```{r, echo=FALSE, out.width="100%", fig.align = "centre", fig.cap="STRING"} 
knitr::include_graphics("https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/string.png")
```

## Gene Ontology

http://geneontology.org/

Gene Ontology (GO) terms are the most widely used set of functional annotations, used by many enrichment tools. The Gene Ontology resource website itself provides several tools for browsing the GO term hierarchy.

<!-- ![](https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/go.png) -->
```{r, echo=FALSE, out.width="100%", fig.align = "centre", fig.cap="Gene Ontology"} 
knitr::include_graphics("https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/go.png")
```

## KEGG

https://www.genome.jp/kegg/

A well-known curated pathway database. It is used by many other tools but with a caveat - KEGG moved to a subscription model in 2011, and so enrichment tools need to use the last open release from 2011. However, up-to-date KEGG pathways are browsable directly through their website. 

<!-- ![](https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/kegg.png) -->
```{r, echo=FALSE, out.width="100%", fig.align = "centre", fig.cap="KEGG"} 
knitr::include_graphics("https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/kegg.png")
```


## GSEA and MSigDB

http://software.broadinstitute.org/gsea/index.jsp

The GSEA Desktop is (one of many) gene set enrichment approaches. It uses gene rankings across all genes rather than hypergeometric or Fisher's exact tests of gene list enrichment. MSigDB (Molecular signatures Database) is a suite of annotation databases suitable for GSEA analysis.

<!-- ![](https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/gsea.png) -->
```{r, echo=FALSE, out.width="100%", fig.align = "centre", fig.cap="MSigDB"} 
knitr::include_graphics("https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/gsea.png")
```


## MetaboAnalyst
MetaboAnalyst is popular among the metabolomics community for statistical, functional, and integrative analyses of metabolomics data. It has features such as **Enrichment Analysis**, **Pathway Analysis**, and **Network Analysis**, which perform metabolite set enrichment, metabolic pathway analysis, and pathway activity prediction from MS peaks.

<!-- ![](https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/metaboanalyst.png) -->
```{r, echo=FALSE, out.width="100%", fig.align = "centre", fig.cap="MetaboAnalyst"} 
knitr::include_graphics("https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/metaboanalyst.png")
```

## Cytoscape

https://cytoscape.org/

Cytoscape is a desktop-based biological network analysis and visualisation tool, rather than a functional enrichment tool (although plugins can extend its functionality). It is mentioned here because it is often useful as a next step when you need to create custom figures showing the interactions of an interesting biological pathway. 

<!-- ![](https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/cytoscape.png) -->
```{r, echo=FALSE, out.width="100%", fig.align = "centre", fig.cap="Cytoscape"} 
knitr::include_graphics("https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/cytoscape.png")
```
