<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 15 clusterProfiler and enrichplot | Functional Enrichment Workshop</title>
<meta name="author" content="Australian Biocommons, Sydney Informatics Hub (USYD) and Monash Genomics and Bioinformatics Platform (MGBP)">
<meta name="description" content="Making sense of gene and proteins lists with functional enrichment analysis  clusterProfiler is a comprehensive suite of enrichment tools. It has functions to run ORA or GSEA over commonly used...">
<meta name="generator" content="bookdown 0.40 with bs4_book()">
<meta property="og:title" content="Chapter 15 clusterProfiler and enrichplot | Functional Enrichment Workshop">
<meta property="og:type" content="book">
<meta property="og:description" content="Making sense of gene and proteins lists with functional enrichment analysis  clusterProfiler is a comprehensive suite of enrichment tools. It has functions to run ORA or GSEA over commonly used...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 15 clusterProfiler and enrichplot | Functional Enrichment Workshop">
<meta name="twitter:description" content="Making sense of gene and proteins lists with functional enrichment analysis  clusterProfiler is a comprehensive suite of enrichment tools. It has functions to run ORA or GSEA over commonly used...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.8.0/transition.js"></script><script src="libs/bs3compat-0.8.0/tabs.js"></script><script src="libs/bs3compat-0.8.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/kePrint-0.0.1/kePrint.js"></script><link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Functional Enrichment Workshop</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> Functional Enrichment Workshop</a></li>
<li class="book-part">Day 1</li>
<li><a class="" href="overview.html"><span class="header-section-number">3</span> Overview</a></li>
<li><a class="" href="recap.html"><span class="header-section-number">4</span> Recap</a></li>
<li><a class="" href="enrichment-statistics.html"><span class="header-section-number">5</span> Enrichment Statistics</a></li>
<li><a class="" href="example-analysis.html"><span class="header-section-number">6</span> Example Analysis</a></li>
<li><a class="" href="defining-the-genelist.html"><span class="header-section-number">7</span> Defining the genelist</a></li>
<li><a class="" href="online-tools.html"><span class="header-section-number">8</span> Online Tools</a></li>
<li><a class="" href="reporting-results.html"><span class="header-section-number">9</span> Reporting results</a></li>
<li><a class="" href="resources.html"><span class="header-section-number">10</span> Resources</a></li>
<li class="book-part">Day 2</li>
<li><a class="" href="introduction.html"><span class="header-section-number">12</span> Introduction</a></li>
<li><a class="" href="r-environment-set-up.html"><span class="header-section-number">13</span> R environment set up</a></li>
<li><a class="" href="gprofiler2.html"><span class="header-section-number">14</span> gprofiler2</a></li>
<li><a class="active" href="clusterprofiler-and-enrichplot.html"><span class="header-section-number">15</span> clusterProfiler and enrichplot</a></li>
<li><a class="" href="webgestaltr.html"><span class="header-section-number">16</span> WebGestaltR</a></li>
<li><a class="" href="novel-species-fea.html"><span class="header-section-number">17</span> Novel species FEA</a></li>
<li><a class="" href="workshop-wrap-up.html"><span class="header-section-number">18</span> Workshop wrap-up</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://monashbioinformaticsplatform.github.io/Functional_Enrichment_Online_Tools/">View book source <i class="fab fa-gitlab"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><span class="main-title"><em>Making sense of gene and proteins lists with functional enrichment analysis</em></span>
<div id="clusterprofiler-and-enrichplot" class="section level1" number="15">
<h1>
<span class="header-section-number">15</span> clusterProfiler and enrichplot<a class="anchor" aria-label="anchor" href="#clusterprofiler-and-enrichplot"><i class="fas fa-link"></i></a>
</h1>
<p><a href="https://bioconductor.org/packages/release/bioc/html/clusterProfiler.html">clusterProfiler</a> is a comprehensive suite of enrichment tools. It has functions to run ORA or GSEA over commonly used databases (GO, KEGG, KEGG Modules, DAVID, Pathway Commons, WikiPathways) as well as universal enrichment functions to perform ORA or GSEA with novel species or custom gene sets. We will use these universal tools in the final activity of this workshop, focusing on the supported organisms and datbases for the present activity.</p>
<p>One of the key advantages of using R over web tools is flexibility with visualisations.</p>
<p>The same authors have released a plotting package <a href="https://www.bioconductor.org/packages/release/bioc/html/enrichplot.html">enrichplot</a> dedicated to plotting enrichment results. In this activity, we will perform GSEA with <code>clusterProfiler</code> then explore many different visualisation options. At the end of the activity, we will have a poll to see which of the many plot types are the favourites! üòÅ</p>
<p>
¬†
</p>
<p><!-- insert blank line --></p>
<div id="supported-databases-species-and-namespaces" class="section level2" number="15.1">
<h2>
<span class="header-section-number">15.1</span> Supported databases, species and namespaces<a class="anchor" aria-label="anchor" href="#supported-databases-species-and-namespaces"><i class="fas fa-link"></i></a>
</h2>
<p>One of the challenges when working with <code>clusterProfiler</code> for FEA is that each enrichment function has different supported organisms and different namespace requirements, so you can not necessarily use all of the functions over the same gene list. In this activity, we will review the FEA functions and investigate their requirements, before performing a gene ID conversion with the <code>bitr</code> function to enable compatability with our <a href="https://link.springer.com/article/10.1007/s10571-016-0403-y">Pezzini et al 2017</a> dataset.</p>
<p>Let‚Äôs start by reviewing the dedicated database enrichment functions and what inbuilt support they have.</p>
<div class="inline-table"><table class="table table-sm">
<colgroup>
<col width="7%">
<col width="15%">
<col width="26%">
<col width="26%">
<col width="11%">
<col width="4%">
<col width="8%">
</colgroup>
<thead><tr class="header">
<th>Database</th>
<th>GO</th>
<th>KEGG</th>
<th>KEGG Modules</th>
<th>Pathway Commons</th>
<th>WikiPathways</th>
<th>DAVID</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>GSEA function</td>
<td>gseGO</td>
<td>gseKEGG</td>
<td>gseMKEGG</td>
<td>gsePC</td>
<td>gseWP</td>
<td>NA</td>
</tr>
<tr class="even">
<td>ORA function</td>
<td>enrichGO</td>
<td>enrichKEGG</td>
<td>enrichMKEGG</td>
<td>enrichPC</td>
<td>enrichWP</td>
<td>enrichDAVID</td>
</tr>
<tr class="odd">
<td>Supported species</td>
<td>Those with Bioconductor annotation package (20)</td>
<td>KEGG Organisms (thousands)</td>
<td>KEGG Organisms (thousands)</td>
<td>Human (or convert to UniProt IDs)</td>
<td>entrez</td>
<td>Those supported by DAVID</td>
</tr>
<tr class="even">
<td>Supported namespaces</td>
<td>differs depending on species</td>
<td>‚Äòkegg‚Äô (compatible with entrez), ‚Äòncbi-geneid‚Äô, ‚Äòncib-proteinid‚Äô or ‚Äòuniprot‚Äô</td>
<td>‚Äòkegg‚Äô (compatible with entrez), ‚Äòncbi-geneid‚Äô, ‚Äòncib-proteinid‚Äô or ‚Äòuniprot‚Äô</td>
<td>‚Äòhgnc‚Äô or ‚Äôuniprot</td>
<td>entrez</td>
<td>Those supported by DAVID</td>
</tr>
</tbody>
</table></div>
<p>To obtain this information, 3 sources were required:</p>
<ol style="list-style-type: decimal">
<li>The <a href="https://bioconductor.org/packages/devel/bioc/manuals/clusterProfiler/man/clusterProfiler.pdf">clusterProfiler user guide</a>
</li>
<li>External websites (<a href="https://bioconductor.org/packages/3.20/data/annotation/">Bioconductor</a> and <a href="https://www.genome.jp/kegg/catalog/org_list.html">KEGG Organisms</a>)</li>
<li>
<code>clusterProfiler</code> functions (eg <code>get_wp_organisms()</code>, <code>keytypes(org.Hs.eg.db)</code>)</li>
</ol>
<p>This highlights the need to carefully review the tool you are using and explore the user guide and functionality.</p>
<p>
¬†
</p>
<p><!-- insert blank line --></p>
</div>
<div id="activity-overview-1" class="section level2" number="15.2">
<h2>
<span class="header-section-number">15.2</span> Activity overview<a class="anchor" aria-label="anchor" href="#activity-overview-1"><i class="fas fa-link"></i></a>
</h2>
<p>Since we have covered ORA with <code>gprofiler</code>, we will perform a GSEA with <code>clusterProfiler</code> using <code>gseKEGG</code>.</p>
<ol style="list-style-type: decimal">
<li>Explore the functions of <code>clusterProfiler</code> including which FEA functions support which organisms and which namespaces</li>
<li>Load input dataset (a gene matrix with adjusted P values and log2 fold change values)</li>
<li>Extract the gene IDs and sort by log2 fold change to create the GSEA ranked gene list R object</li>
<li>Use <code>bitr</code> to convert gene IDs from ENSEMBL to ENTREZ for compatability with <code>gseKEGG</code>
</li>
<li>Perform GSEA with <code>gseKEGG</code>
</li>
<li>Visualise results with many different plot types from <code>enrichplot</code>
</li>
</ol>
<p>
¬†
</p>
<p><!-- insert blank line --></p>
<p>Let‚Äôs head over to RStudio now and try out some functions! üèÉ</p>
<p>
¬†
</p>
<p><!-- insert blank line --></p>
<p>‚û§ Go back to your RStudio interface and clear your environment by selecting <code>Session</code> ‚Üí <code>Quit session</code> ‚Üí <code>Dont save</code> ‚Üí <code>Start new session</code></p>
<p>‚û§ Open the <code>clusterProfiler.Rmd</code> notebook</p>
<p><strong><span style="color: #006400;">Instructions for the analysis will continue from the R notebook.</span></strong></p>
<p>
¬†
</p>
<p><!-- insert blank line --></p>
</div>
<div id="end-of-activity-summary-1" class="section level2" number="15.3">
<h2>
<span class="header-section-number">15.3</span> End of activity summary<a class="anchor" aria-label="anchor" href="#end-of-activity-summary-1"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li>We have explored the supported organisms, namespaces and databases of the <code>clusterProfiler</code> enrichment functions</li>
<li>We have extracted a ranked gene list for GSEA and converted the gene IDs for compatability with <code>gseKEGG</code>
</li>
<li>We have performed GSEA on the KEGG database with <code>gseKEGG</code> and visualised the results with multiple plot types</li>
<li>We have captured all version details relevant to the session within the R notebook and knit the file to HTML for record keeping</li>
</ul>
<p>
¬†
</p>
<p><!-- insert blank line --></p>
</div>
<div id="poll" class="section level2" number="15.4">
<h2>
<span class="header-section-number">15.4</span> Poll<a class="anchor" aria-label="anchor" href="#poll"><i class="fas fa-link"></i></a>
</h2>
<p>‚ùìWhat was your favourite R plot from this activity? ü§î</p>
<p>This may be the one you found most informative, easiest to interpret, most eye-catching‚Ä¶</p>
<p>
¬†
</p>
<p><!-- insert blank line --></p>
<div class="inline-figure"><img src="images/poll-favourite-plot-type.png" style="border: none; box-shadow: none; background: none; width: 100%;"></div>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="gprofiler2.html"><span class="header-section-number">14</span> gprofiler2</a></div>
<div class="next"><a href="webgestaltr.html"><span class="header-section-number">16</span> WebGestaltR</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#clusterprofiler-and-enrichplot"><span class="header-section-number">15</span> clusterProfiler and enrichplot</a></li>
<li><a class="nav-link" href="#supported-databases-species-and-namespaces"><span class="header-section-number">15.1</span> Supported databases, species and namespaces</a></li>
<li><a class="nav-link" href="#activity-overview-1"><span class="header-section-number">15.2</span> Activity overview</a></li>
<li><a class="nav-link" href="#end-of-activity-summary-1"><span class="header-section-number">15.3</span> End of activity summary</a></li>
<li><a class="nav-link" href="#poll"><span class="header-section-number">15.4</span> Poll</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://monashbioinformaticsplatform.github.io/Functional_Enrichment_Online_Tools//blob/master/12-clusterprofiler.Rmd">View source <i class="fab fa-gitlab"></i></a></li>
          <li><a id="book-edit" href="https://monashbioinformaticsplatform.github.io/Functional_Enrichment_Online_Tools//edit/master/12-clusterprofiler.Rmd">Edit this page <i class="fab fa-gitlab"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Functional Enrichment Workshop</strong>" was written by Australian Biocommons, Sydney Informatics Hub (USYD) and Monash Genomics and Bioinformatics Platform (MGBP). It was last built on <span>‚ÄúOctober 14, 2025‚Äù</span> <br><img src="logos.png" width="300" alt="Logo"><br>.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
