<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 6 Defining the genelist | Functional Enrichment Workshop</title>
<meta name="author" content="Australian Biocommons, Sydney Informatics Hub (USYD) and Monash Genomics and Bioinformatics Platform (MGBP)">
<meta name="description" content="Making sense of gene and proteins lists with functional enrichment analysis  Starting from those differential expression results here, how do we go about getting a genelist to calculate enrichment...">
<meta name="generator" content="bookdown 0.40 with bs4_book()">
<meta property="og:title" content="Chapter 6 Defining the genelist | Functional Enrichment Workshop">
<meta property="og:type" content="book">
<meta property="og:description" content="Making sense of gene and proteins lists with functional enrichment analysis  Starting from those differential expression results here, how do we go about getting a genelist to calculate enrichment...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 6 Defining the genelist | Functional Enrichment Workshop">
<meta name="twitter:description" content="Making sense of gene and proteins lists with functional enrichment analysis  Starting from those differential expression results here, how do we go about getting a genelist to calculate enrichment...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.8.0/transition.js"></script><script src="libs/bs3compat-0.8.0/tabs.js"></script><script src="libs/bs3compat-0.8.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/kePrint-0.0.1/kePrint.js"></script><link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Functional Enrichment Workshop</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> Functional Enrichment Workshop</a></li>
<li><a class="" href="overview.html"><span class="header-section-number">2</span> Overview</a></li>
<li><a class="" href="recap.html"><span class="header-section-number">3</span> Recap</a></li>
<li><a class="" href="enrichment-statistics.html"><span class="header-section-number">4</span> Enrichment Statistics</a></li>
<li><a class="" href="example-analysis.html"><span class="header-section-number">5</span> Example Analysis</a></li>
<li><a class="active" href="defining-the-genelist.html"><span class="header-section-number">6</span> Defining the genelist</a></li>
<li><a class="" href="online-tools.html"><span class="header-section-number">7</span> Online Tools</a></li>
<li><a class="" href="reporting-results.html"><span class="header-section-number">8</span> Reporting results</a></li>
<li><a class="" href="resources.html"><span class="header-section-number">9</span> Resources</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://monashbioinformaticsplatform.github.io/Functional_Enrichment_Online_Tools/">View book source <i class="fab fa-gitlab"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><span class="main-title"><em>Making sense of gene and proteins lists with functional enrichment analysis</em></span>
<div id="defining-the-genelist" class="section level1" number="6">
<h1>
<span class="header-section-number">6</span> Defining the genelist<a class="anchor" aria-label="anchor" href="#defining-the-genelist"><i class="fas fa-link"></i></a>
</h1>
<p>Starting from those differential expression results <a href="http://degust.erc.monash.edu/degust/compare.html?code=5b2c7805ab8f8c5f2dc8c72e61b049b0#?plot=mds">here</a>, how do we go about getting a genelist to calculate enrichment on?</p>
<div id="activities" class="section level2" number="6.1">
<h2>
<span class="header-section-number">6.1</span> Activities<a class="anchor" aria-label="anchor" href="#activities"><i class="fas fa-link"></i></a>
</h2>
<p>Todays exercise follows the process of getting the differentially expressed gene list using excel. You could use another spreadsheet program, or some may prefer a programming language like R .</p>
<ol style="list-style-type: decimal">
<li><p>Download the full table of data from either degust, or the csv file here:
<a href="https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/data/Pezzini2016_SHSY5Ycelldiff_DE_table.csv">Pezzini2016_SHSY5Ycelldiff_DE_table.csv</a>. Import into excel. <!-- File>Import --></p></li>
<li>
<p>How many genes are differentially expressed? In these results the FDR Column contains the corrected p-value, and the ‘differentiated’ column shows the log2 fold-change of differentiated cells vs untreated cells (log2(diff)-log2(undiff)); 0 is unchanged, 1 is doubled, -1 is halved.</p>
<ul>
<li><p>Significant at 0.01?</p></li>
<li><p>That’s a particularly large number of genes - perhaps not unexpected given how much the cells are changed this experiment. How many significant genes also have 2-fold change in expression?</p></li>
<li><p>For this workshop, get the genes with a FDR&lt;0.01 and 2x fold change (<code>log2(4)</code>). Note - most experiments yeild far less differential expression, but the difference between these two cell conditions is pretty extreme! Typically you would only filter at p&lt;0.01 (and occasionally 2-fold change) - you might see 10s to 100s of results. However, this arbitrary threshold gives a more typical number of differentially expressed genes for downstream analysis. An alternative approach could be to take the top 500 genes.</p></li>
</ul>
</li>
</ol>
<details><summary>
Show
</summary>
There are 4923 differentially expressed genes, 2149 of which have a 2-fold change in expression. With the aggressive filtering, there are 792 genes left.
</details><ol start="3" style="list-style-type: decimal">
<li>How many genes are <em>tested</em>? This is your background.</li>
</ol>
<details><summary>
Show
</summary>
14420 genes tested.
</details><!--But with ~20k human genes - why are there genes missing? **14420** --><hr>
</div>
<div id="common-gotcha" class="section level2" number="6.2">
<h2>
<span class="header-section-number">6.2</span> Common gotcha<a class="anchor" aria-label="anchor" href="#common-gotcha"><i class="fas fa-link"></i></a>
</h2>
<p>Can you find SEPT4? Because <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-1044-7"><em>Gene name errors are widespread in the scientific literature</em></a></p>
<p>You can’t revert the gene names automatically (try converting it to text!). You have to avoid it in the first place by importing gene columns as ‘text’ columns in excel. See video from HUGO : <a href="https://www.genenames.org/help/faq/#!/#tocAnchor-1-25-1" class="uri">https://www.genenames.org/help/faq/#!/#tocAnchor-1-25-1</a></p>
<!--NB: You can ignore these for this workshop, but you want this to be right for publication!-->
<hr>
</div>
<div id="example" class="section level2" number="6.3">
<h2>
<span class="header-section-number">6.3</span> Example<a class="anchor" aria-label="anchor" href="#example"><i class="fas fa-link"></i></a>
</h2>
<p>An example excel document showing this filtering process is here: <a href="https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/data/Pezzini2016_SHSY5Ycelldiff_DE_table_filtering.xlsx">Pezzini2016_SHSY5Ycelldiff_DE_table_filtering.xlsx</a>.</p>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="example-analysis.html"><span class="header-section-number">5</span> Example Analysis</a></div>
<div class="next"><a href="online-tools.html"><span class="header-section-number">7</span> Online Tools</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#defining-the-genelist"><span class="header-section-number">6</span> Defining the genelist</a></li>
<li><a class="nav-link" href="#activities"><span class="header-section-number">6.1</span> Activities</a></li>
<li><a class="nav-link" href="#common-gotcha"><span class="header-section-number">6.2</span> Common gotcha</a></li>
<li><a class="nav-link" href="#example"><span class="header-section-number">6.3</span> Example</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://monashbioinformaticsplatform.github.io/Functional_Enrichment_Online_Tools//blob/master/05-genelists.Rmd">View source <i class="fab fa-gitlab"></i></a></li>
          <li><a id="book-edit" href="https://monashbioinformaticsplatform.github.io/Functional_Enrichment_Online_Tools//edit/master/05-genelists.Rmd">Edit this page <i class="fab fa-gitlab"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Functional Enrichment Workshop</strong>" was written by Australian Biocommons, Sydney Informatics Hub (USYD) and Monash Genomics and Bioinformatics Platform (MGBP). It was last built on <span>“October 31, 2025”</span> <br><img src="logos.png" width="300" alt="Logo"><br>.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
