<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 4 Enrichment Statistics | Functional Enrichment Workshop</title>
<meta name="author" content="Australian Biocommons, Sydney Informatics Hub (USYD) and Monash Genomics and Bioinformatics Platform (MGBP)">
<meta name="description" content="Making sense of gene and proteins lists with functional enrichment analysis  Enrichment statistics are based on a contingency table like so:     ..in term  ..not in term  Total  ..in gene list  50...">
<meta name="generator" content="bookdown 0.40 with bs4_book()">
<meta property="og:title" content="Chapter 4 Enrichment Statistics | Functional Enrichment Workshop">
<meta property="og:type" content="book">
<meta property="og:description" content="Making sense of gene and proteins lists with functional enrichment analysis  Enrichment statistics are based on a contingency table like so:     ..in term  ..not in term  Total  ..in gene list  50...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 4 Enrichment Statistics | Functional Enrichment Workshop">
<meta name="twitter:description" content="Making sense of gene and proteins lists with functional enrichment analysis  Enrichment statistics are based on a contingency table like so:     ..in term  ..not in term  Total  ..in gene list  50...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.8.0/transition.js"></script><script src="libs/bs3compat-0.8.0/tabs.js"></script><script src="libs/bs3compat-0.8.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/kePrint-0.0.1/kePrint.js"></script><link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Functional Enrichment Workshop</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> Functional Enrichment Workshop</a></li>
<li><a class="" href="overview.html"><span class="header-section-number">2</span> Overview</a></li>
<li><a class="" href="recap.html"><span class="header-section-number">3</span> Recap</a></li>
<li><a class="active" href="enrichment-statistics.html"><span class="header-section-number">4</span> Enrichment Statistics</a></li>
<li><a class="" href="example-analysis.html"><span class="header-section-number">5</span> Example Analysis</a></li>
<li><a class="" href="defining-the-genelist.html"><span class="header-section-number">6</span> Defining the genelist</a></li>
<li><a class="" href="online-tools.html"><span class="header-section-number">7</span> Online Tools</a></li>
<li><a class="" href="reporting-results.html"><span class="header-section-number">8</span> Reporting results</a></li>
<li><a class="" href="resources.html"><span class="header-section-number">9</span> Resources</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://monashbioinformaticsplatform.github.io/Functional_Enrichment_Online_Tools/">View book source <i class="fab fa-gitlab"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><span class="main-title"><em>Making sense of gene and proteins lists with functional enrichment analysis</em></span>
<div id="enrichment-statistics" class="section level1" number="4">
<h1>
<span class="header-section-number">4</span> Enrichment Statistics<a class="anchor" aria-label="anchor" href="#enrichment-statistics"><i class="fas fa-link"></i></a>
</h1>
<p>Enrichment statistics are based on a contingency table like so:</p>
<!-- |        Gene        | ..in term | ..not in term | Total | -->
<!-- |:------------------:|----------:|--------------:|------:| -->
<!-- |     ..in gene list |        50 |           100 |   150 | -->
<!-- | ..not in gene list (but in background) |       200 |         15900 | 16100 | -->
<!-- |              Total |       250 |         16000 | 16250 | -->
<div class="inline-table"><table border="1" style="border-collapse: collapse; width: 100%;" class="table table-sm">
<tr>
<th>
</th>
<th>
..in term
</th>
<th>
..not in term
</th>
<td>
Total
</td>
</tr>
<tr>
<th>
..in gene list
</th>
<td style="background-color: #ffcccb;">
50
</td>
<td style="background-color: #90ee90;">
100
</td>
<td>
150
</td>
</tr>
<tr>
<th>
..not in gene list (but in background)
</th>
<td style="background-color: #add8e6;">
200
</td>
<td style="background-color: #ffffe0;">
15900
</td>
<td>
16100
</td>
</tr>
<tr>
<td>
Total
</td>
<td>
250
</td>
<td>
16000
</td>
<!-- <td>16250</td> -->
<td style="color: orange;">
16250
</td>
<!-- Colored text for 16250 -->
</tr>
</table></div>
<p>This is based on the 16250 genes that were measured in your experiment.</p>
<p>Note that there might be extra genes that weren’t measured these are excluded from the calculations entirely. E.g. There might have been an extra 5000 terms (some of which might have been annotated with the term of interest), making for 21250 <em>annotated</em> genes.</p>
<hr>
<div id="fishers-exact-test" class="section level2" number="4.1">
<h2>
<span class="header-section-number">4.1</span> <span style="color:orange;">Fisher’s Exact Test</span><a class="anchor" aria-label="anchor" href="#fishers-exact-test"><i class="fas fa-link"></i></a>
</h2>
<p>Fisher’s Exact Test is a statistical test used to determine if there are nonrandom associations between the proportions of two categorical variables. It calculates the exact probability of observing the given distribution of counts in a 2x2 contingency table, under the null hypothesis of no association between the variables.</p>
<blockquote>
<p><em>Note:</em> This is just a toy calculator for this training, it is quite limited. You can also use some online tools like <a href="https://www.socscistatistics.com/tests/fisher/default2.aspx">Social Science Statistics</a> to play with.</p>
</blockquote>
<p><em>Formula</em>:
<span class="math display">\[P = \frac{(a + b)!(c + d)!(a + c)!(b + d)!}{a!b!c!d!N!}\]</span></p>
<p>Where:</p>
<ul>
<li><p><span class="math inline">\(a\)</span>, <span class="math inline">\(b\)</span>, <span class="math inline">\(c\)</span>, and <span class="math inline">\(d\)</span> are the observed counts in the 2x2 contingency table.</p></li>
<li><p><span class="math inline">\(N\)</span> is the total number of observations, <span class="math inline">\(N = a + b + c + d\)</span>.</p></li>
</ul>
<p>Given this contingency table:</p>
<div class="inline-table"><table class="table table-sm">
<thead><tr class="header">
<th></th>
<th>Category 1</th>
<th>Category 2</th>
<th>Total</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><strong>Group 1</strong></td>
<td><span class="math inline">\(a\)</span></td>
<td><span class="math inline">\(b\)</span></td>
<td><span class="math inline">\(a + b\)</span></td>
</tr>
<tr class="even">
<td><strong>Group 2</strong></td>
<td><span class="math inline">\(c\)</span></td>
<td><span class="math inline">\(d\)</span></td>
<td><span class="math inline">\(c + d\)</span></td>
</tr>
<tr class="odd">
<td><strong>Total</strong></td>
<td><span class="math inline">\(a + c\)</span></td>
<td><span class="math inline">\(b + d\)</span></td>
<td><span class="math inline">\(a + b + c + d\)</span></td>
</tr>
</tbody>
</table></div>
<p><em>R syntax</em></p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">a</span><span class="op">=</span><span class="fl">50</span></span>
<span><span class="va">b</span><span class="op">=</span><span class="fl">100</span></span>
<span><span class="va">c</span><span class="op">=</span><span class="fl">200</span></span>
<span><span class="va">d</span><span class="op">=</span><span class="fl">15900</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">a</span>, <span class="va">b</span>, <span class="va">c</span>, <span class="va">d</span><span class="op">)</span>, nrow <span class="op">=</span> <span class="fl">2</span>, byrow <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># One-tailed test</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/stats/fisher.test.html">fisher.test</a></span><span class="op">(</span><span class="va">data</span>, alternative <span class="op">=</span> <span class="st">"g"</span><span class="op">)</span> <span class="co"># greater or "less"</span></span></code></pre></div>
</div>
<div id="hypergeometric-test" class="section level2" number="4.2">
<h2>
<span class="header-section-number">4.2</span> <span style="color:orange;">Hypergeometric Test</span><a class="anchor" aria-label="anchor" href="#hypergeometric-test"><i class="fas fa-link"></i></a>
</h2>
<p>Hypergeometric test calculates the probability of observing the given number of genes from a specific category (e.g., a pathway) in the gene list (differentially expressed genes) by chance. It models the situation where you draw a sample (the gene list) from a finite population (the background of all genes), and success is defined as a gene being in the category (e.g., belonging to the pathway).</p>
<blockquote>
<p><em>Note:</em> Here is a tool by <a href="https://stattrek.com/online-calculator/hypergeometric">Stat Trek</a> to play around with the hypergeometric test.</p>
</blockquote>
<p><em>Formula:</em>
<span class="math display">\[P(X = k) = \frac{\binom{K}{k} \binom{N - K}{n - k}}{\binom{N}{n}}\]</span>
Where:</p>
<ul>
<li><p><span class="math inline">\(N\)</span> = Total number of items in the population.</p></li>
<li><p><span class="math inline">\(K\)</span> = Number of success items in the population.</p></li>
<li><p><span class="math inline">\(n\)</span> = Number of items in the sample.</p></li>
<li><p><span class="math inline">\(k\)</span> = Number of success items in the sample.</p></li>
</ul>
<p>The parameters in our example:
N=16250; K=250; n=150; k=50</p>
<p><em>R syntax</em></p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Observed value</span></span>
<span><span class="va">N</span> <span class="op">&lt;-</span> <span class="va">a</span> <span class="op">+</span> <span class="va">b</span> <span class="op">+</span> <span class="va">c</span> <span class="op">+</span> <span class="va">d</span></span>
<span><span class="va">K</span> <span class="op">&lt;-</span> <span class="va">a</span> <span class="op">+</span> <span class="va">c</span></span>
<span><span class="va">n</span> <span class="op">&lt;-</span> <span class="va">a</span> <span class="op">+</span> <span class="va">b</span></span>
<span></span>
<span><span class="co"># One-tailed p-value (P[X &gt;= a])</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/stats/Hypergeometric.html">phyper</a></span><span class="op">(</span><span class="va">a</span> <span class="op">-</span> <span class="fl">1</span>, <span class="va">K</span>, <span class="va">N</span> <span class="op">-</span> <span class="va">K</span>, <span class="va">n</span>, lower.tail <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>Where:</p>
<p>k−1 is the number of observed successes minus 1 (for the “at least” scenario).
lower.tail = FALSE gives the probability of getting at least k successes (right-tail).</p>
<hr>
<div id="section" class="section level4 unnumbered">
<h4 class="unnumbered"><a class="anchor" aria-label="anchor" href="#section"><i class="fas fa-link"></i></a></h4>
</div>
</div>
<div id="activity" class="section level2" number="4.3">
<h2>
<span class="header-section-number">4.3</span> Activity<a class="anchor" aria-label="anchor" href="#activity"><i class="fas fa-link"></i></a>
</h2>
<div id="challenge-interactive-calculator" class="section level3 unnumbered challenge">
<h3>
<strong>Challenge:</strong> Interactive Calculator<a class="anchor" aria-label="anchor" href="#challenge-interactive-calculator"><i class="fas fa-link"></i></a>
</h3>
<p><a href="https://bioinformatics3.erc.monash.edu/rsconnect/content/241/"><em>Link to open toy enrichment calculator</em></a>.</p>
<p>This calculates enrichment for a single hypothetical genelist (e.g. your RNAseq differentially expressed genelist) against a single hypothetical ‘term’ (some set of interesting genes, e.g. synaptic signaling genes). It makes a Venn diagram and a wordy description of what is being tested.</p>
<p>You can adjust various factors and see their effect on the enrichment p-values.</p>
<div id="section-1" class="section level4 unnumbered">
<h4 class="unnumbered"><a class="anchor" aria-label="anchor" href="#section-1"><i class="fas fa-link"></i></a></h4>
</div>
</div>
<div id="questions" class="section level3 unnumbered rationale">
<h3>
<strong>Questions</strong><a class="anchor" aria-label="anchor" href="#questions"><i class="fas fa-link"></i></a>
</h3>
<ol style="list-style-type: decimal">
<li>
<p>If 24 of the 300 differentially expressed genes are annotated with the 500-gene term of interest. Is it significant at p=0.05?</p>
<details><summary><p>Show</p>
</summary><p>No, corrected pval=0.087</p>
</details><p>
</p>
<p> </p>

<p><!-- insert blank line --></p>
</li>
<li>
<p>What about with a smaller background of 5000 genes (e.g. proteomic datasets)?</p>
<details><summary><p>Show</p>
</summary><p>Even less so - corrected pval=1</p>
</details><p>
</p>
<p> </p>

<p><!-- insert blank line --></p>
</li>
<li>
<p>Or, testing against a smaller database of terms; 2000 terms instead of 10000? With the original 16000 gene background.</p>
<details><summary><p>Show</p>
</summary><p>Yes, now corrected pval=0.017</p>
</details><p>
</p>
<p> </p>

<p><!-- insert blank line --></p>
</li>
<li>
<p>19 out of 200 differentially expressed genes (9.5%), need to hit for a 500-gene term (3.1% of all genes) to be significant at (p=0.048). How many hits would be needed for a more specific 30-gene term?</p>
<details><summary><p>Show</p>
</summary><p>5 hits - 2.5% of the differentially expressed genes vs 0.19% of all genes</p>
</details>
</li>
</ol>
<div id="section-2" class="section level4 unnumbered">
<h4 class="unnumbered"><a class="anchor" aria-label="anchor" href="#section-2"><i class="fas fa-link"></i></a></h4>

</div>
</div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="recap.html"><span class="header-section-number">3</span> Recap</a></div>
<div class="next"><a href="example-analysis.html"><span class="header-section-number">5</span> Example Analysis</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#enrichment-statistics"><span class="header-section-number">4</span> Enrichment Statistics</a></li>
<li><a class="nav-link" href="#fishers-exact-test"><span class="header-section-number">4.1</span> Fisher’s Exact Test</a></li>
<li><a class="nav-link" href="#hypergeometric-test"><span class="header-section-number">4.2</span> Hypergeometric Test</a></li>
<li>
<a class="nav-link" href="#activity"><span class="header-section-number">4.3</span> Activity</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#challenge-interactive-calculator">Challenge: Interactive Calculator</a></li>
<li><a class="nav-link" href="#questions">Questions</a></li>
</ul>
</li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://monashbioinformaticsplatform.github.io/Functional_Enrichment_Online_Tools//blob/master/03-stats.Rmd">View source <i class="fab fa-gitlab"></i></a></li>
          <li><a id="book-edit" href="https://monashbioinformaticsplatform.github.io/Functional_Enrichment_Online_Tools//edit/master/03-stats.Rmd">Edit this page <i class="fab fa-gitlab"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Functional Enrichment Workshop</strong>" was written by Australian Biocommons, Sydney Informatics Hub (USYD) and Monash Genomics and Bioinformatics Platform (MGBP). It was last built on <span>“October 16, 2025”</span> <br><img src="logos.png" width="300" alt="Logo"><br>.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
