<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 14 gprofiler2 | Functional Enrichment Workshop</title>
<meta name="author" content="Australian Biocommons, Sydney Informatics Hub (USYD) and Monash Genomics and Bioinformatics Platform (MGBP)">
<meta name="description" content="Making sense of gene and proteins lists with functional enrichment analysis  gprofiler2 is the R interface to the g:Profiler web-based toolset that you used in day 1 of the workshop. Like the web...">
<meta name="generator" content="bookdown 0.40 with bs4_book()">
<meta property="og:title" content="Chapter 14 gprofiler2 | Functional Enrichment Workshop">
<meta property="og:type" content="book">
<meta property="og:description" content="Making sense of gene and proteins lists with functional enrichment analysis  gprofiler2 is the R interface to the g:Profiler web-based toolset that you used in day 1 of the workshop. Like the web...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 14 gprofiler2 | Functional Enrichment Workshop">
<meta name="twitter:description" content="Making sense of gene and proteins lists with functional enrichment analysis  gprofiler2 is the R interface to the g:Profiler web-based toolset that you used in day 1 of the workshop. Like the web...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.8.0/transition.js"></script><script src="libs/bs3compat-0.8.0/tabs.js"></script><script src="libs/bs3compat-0.8.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/kePrint-0.0.1/kePrint.js"></script><link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Functional Enrichment Workshop</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> Functional Enrichment Workshop</a></li>
<li class="book-part">Day 1</li>
<li><a class="" href="overview.html"><span class="header-section-number">3</span> Overview</a></li>
<li><a class="" href="recap.html"><span class="header-section-number">4</span> Recap</a></li>
<li><a class="" href="enrichment-statistics.html"><span class="header-section-number">5</span> Enrichment Statistics</a></li>
<li><a class="" href="example-analysis.html"><span class="header-section-number">6</span> Example Analysis</a></li>
<li><a class="" href="defining-the-genelist.html"><span class="header-section-number">7</span> Defining the genelist</a></li>
<li><a class="" href="online-tools.html"><span class="header-section-number">8</span> Online Tools</a></li>
<li><a class="" href="reporting-results.html"><span class="header-section-number">9</span> Reporting results</a></li>
<li><a class="" href="resources.html"><span class="header-section-number">10</span> Resources</a></li>
<li class="book-part">Day 2</li>
<li><a class="" href="introduction.html"><span class="header-section-number">12</span> Introduction</a></li>
<li><a class="" href="r-environment-set-up.html"><span class="header-section-number">13</span> R environment set up</a></li>
<li><a class="active" href="gprofiler2.html"><span class="header-section-number">14</span> gprofiler2</a></li>
<li><a class="" href="clusterprofiler-and-enrichplot.html"><span class="header-section-number">15</span> clusterProfiler and enrichplot</a></li>
<li><a class="" href="webgestaltr.html"><span class="header-section-number">16</span> WebGestaltR</a></li>
<li><a class="" href="novel-species-fea.html"><span class="header-section-number">17</span> Novel species FEA</a></li>
<li><a class="" href="workshop-wrap-up.html"><span class="header-section-number">18</span> Workshop wrap-up</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://monashbioinformaticsplatform.github.io/Functional_Enrichment_Online_Tools/">View book source <i class="fab fa-gitlab"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><span class="main-title"><em>Making sense of gene and proteins lists with functional enrichment analysis</em></span>
<div id="gprofiler2" class="section level1" number="14">
<h1>
<span class="header-section-number">14</span> gprofiler2<a class="anchor" aria-label="anchor" href="#gprofiler2"><i class="fas fa-link"></i></a>
</h1>
<p><a href="https://cran.r-project.org/web/packages/gprofiler2/index.html">gprofiler2</a> is the R interface to the <code>g:Profiler</code> web-based toolset that you used in day 1 of the workshop.</p>
<p>Like the web interface, <code>gprofiler2</code> performs ORA with <code>g:GOSt</code> against multiple databases simultaneously.</p>
<p>It supports all the same organisms, namespaces and data sources as the web tool. The list of organisms and corresponding data sources is available <a href="https://biit.cs.ut.ee/gprofiler/page/organism-list">here</a> (n = 984).</p>
<p>The full list of supported namespaces is available <a href="https://biit.cs.ut.ee/gprofiler/page/namespaces-list">here</a>.</p>
<p>The <code>gprofiler2</code> user guide can be found <a href="https://cran.r-project.org/web/packages/gprofiler2/gprofiler2.pdf">here</a>.</p>
<p>
 
</p>
<p><!-- insert blank line --></p>
<div id="input-data-3" class="section level2" number="14.1">
<h2>
<span class="header-section-number">14.1</span> Input data<a class="anchor" aria-label="anchor" href="#input-data-3"><i class="fas fa-link"></i></a>
</h2>
<p>Since we are doing ORA, we will need a filtered gene list, and a background gene list. We will continue with the RNAseq dataset from <a href="https://link.springer.com/article/10.1007/s10571-016-0403-y">Pezzini et al 2017</a> introduced yesterday.</p>
<p>
 
</p>
<p><!-- insert blank line --></p>
</div>
<div id="activity-overview" class="section level2" number="14.2">
<h2>
<span class="header-section-number">14.2</span> Activity overview<a class="anchor" aria-label="anchor" href="#activity-overview"><i class="fas fa-link"></i></a>
</h2>
<ol style="list-style-type: decimal">
<li>Load input dataset (a gene matrix with adjusted P values and log2 fold change values)</li>
<li>Filter for differentially expressed genes (DEGs) and create a gene list R object</li>
<li>Extract background genes and create a background gene list R object</li>
<li>Run ORA with <code>gost</code> function</li>
<li>Save the tabular results to a file</li>
<li>Visualise the results with <code>gostplot</code>
</li>
<li>Run a <code>gost</code> multi-query for up-regulated and down-regulated genes</li>
<li>Compare <code>gprofiler2</code> R results to the <code>g:Profiler</code> web results</li>
</ol>
<p>
 
</p>
<p><!-- insert blank line --></p>
<p>➤ Go back to your RStudio interface, where we have opened the <code>gprofiler2.Rmd</code> notebook and loaded the required R packages.</p>
<p><strong><span style="color: #006400;">Instructions for the analysis will continue from the R notebook.</span></strong></p>
<p>
 
</p>
<p><!-- insert blank line --></p>
<p><strong>To finish up:</strong> The last task is to <code>knit</code> the notebook. Our notebook is editable, and can be changed. Deleting code deletes the output, so we could lose valuable details. If we knit the notebook to HTML, we have a permanent static copy of the work.</p>
<p>➤ Knit the notebook to HTML</p>
<p>On the <code>editor</code> pane toolbar, under <code>Preview</code>, select <code>Knit to HTML</code>.</p>
<p>If you have already run <code>Preview</code>, you will see <code>Knit</code> instead of <code>Preview</code>.</p>
<p>The HTML file will be saved in the same directory as the notebook, and with the same filename, but the <code>.Rmd</code> prefix will be replaced by <code>.html</code>. The knit HTML will typically open automatically once complete.</p>
<p>❗If you receive a popup blocker error, click <code>cancel</code> (not <code>try again</code>), and in the <code>Files</code> pane of RStudio, single click the <code>gprofiler.html</code> file and select <code>View in Web Browser</code>.</p>
<div class="inline-figure"><img src="images/open-html-rstudio.png" style="border: none; box-shadow: none; background: none; "></div>
<p>
 
</p>
<p><!-- insert blank line --></p>
<p>Note that the notebook will only successsfully knit if there are <span style="color: red;">no errors</span> in the code. You can ‘preview’ HTML with code errors.</p>
<p>✋ <strong>If your knit fails, please ask for assistance resolving the errors</strong></p>
</div>
<div id="end-of-activity-summary" class="section level2" number="14.3">
<h2>
<span class="header-section-number">14.3</span> End of activity summary<a class="anchor" aria-label="anchor" href="#end-of-activity-summary"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li>We have extracted a gene list and background gene list from a DE dataset and run ORA with <code>gprofiler2</code> <code>gost</code> function</li>
<li>We have plotted the data with <code>gostplot</code> Manhattan plots and <code>ggplot2</code> dotplots</li>
<li>We have run a <code>gost</code> multi-query separating up and down regulated genes</li>
<li>We have verified that <code>gprofiler2</code> results match the results from <code>g:Profiler</code> web</li>
<li>We have saved a rendered HTML document as a static record of our analysis including all code executed, parameters applied, as well as version information for R, RStudio, R packages and gProfiler database to ensure reproducibility</li>
</ul>
<p>
 
</p>
<p><!-- insert blank line --></p>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="r-environment-set-up.html"><span class="header-section-number">13</span> R environment set up</a></div>
<div class="next"><a href="clusterprofiler-and-enrichplot.html"><span class="header-section-number">15</span> clusterProfiler and enrichplot</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#gprofiler2"><span class="header-section-number">14</span> gprofiler2</a></li>
<li><a class="nav-link" href="#input-data-3"><span class="header-section-number">14.1</span> Input data</a></li>
<li><a class="nav-link" href="#activity-overview"><span class="header-section-number">14.2</span> Activity overview</a></li>
<li><a class="nav-link" href="#end-of-activity-summary"><span class="header-section-number">14.3</span> End of activity summary</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://monashbioinformaticsplatform.github.io/Functional_Enrichment_Online_Tools//blob/master/11-gprofiler2.Rmd">View source <i class="fab fa-gitlab"></i></a></li>
          <li><a id="book-edit" href="https://monashbioinformaticsplatform.github.io/Functional_Enrichment_Online_Tools//edit/master/11-gprofiler2.Rmd">Edit this page <i class="fab fa-gitlab"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Functional Enrichment Workshop</strong>" was written by Australian Biocommons, Sydney Informatics Hub (USYD) and Monash Genomics and Bioinformatics Platform (MGBP). It was last built on <span>“October 14, 2025”</span> <br><img src="logos.png" width="300" alt="Logo"><br>.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
