<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 12 Introduction | Functional Enrichment Workshop</title>
<meta name="author" content="Australian Biocommons, Sydney Informatics Hub (USYD) and Monash Genomics and Bioinformatics Platform (MGBP)">
<meta name="description" content="Making sense of gene and proteins lists with functional enrichment analysis  For today’s session, we will be using the R statistical programming language to perform functional enrichmemt analysis...">
<meta name="generator" content="bookdown 0.40 with bs4_book()">
<meta property="og:title" content="Chapter 12 Introduction | Functional Enrichment Workshop">
<meta property="og:type" content="book">
<meta property="og:description" content="Making sense of gene and proteins lists with functional enrichment analysis  For today’s session, we will be using the R statistical programming language to perform functional enrichmemt analysis...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 12 Introduction | Functional Enrichment Workshop">
<meta name="twitter:description" content="Making sense of gene and proteins lists with functional enrichment analysis  For today’s session, we will be using the R statistical programming language to perform functional enrichmemt analysis...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.8.0/transition.js"></script><script src="libs/bs3compat-0.8.0/tabs.js"></script><script src="libs/bs3compat-0.8.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/kePrint-0.0.1/kePrint.js"></script><link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Functional Enrichment Workshop</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> Functional Enrichment Workshop</a></li>
<li class="book-part">Day 1</li>
<li><a class="" href="overview.html"><span class="header-section-number">3</span> Overview</a></li>
<li><a class="" href="recap.html"><span class="header-section-number">4</span> Recap</a></li>
<li><a class="" href="enrichment-statistics.html"><span class="header-section-number">5</span> Enrichment Statistics</a></li>
<li><a class="" href="example-analysis.html"><span class="header-section-number">6</span> Example Analysis</a></li>
<li><a class="" href="defining-the-genelist.html"><span class="header-section-number">7</span> Defining the genelist</a></li>
<li><a class="" href="online-tools.html"><span class="header-section-number">8</span> Online Tools</a></li>
<li><a class="" href="reporting-results.html"><span class="header-section-number">9</span> Reporting results</a></li>
<li><a class="" href="resources.html"><span class="header-section-number">10</span> Resources</a></li>
<li class="book-part">Day 2</li>
<li><a class="active" href="introduction.html"><span class="header-section-number">12</span> Introduction</a></li>
<li><a class="" href="r-environment-set-up.html"><span class="header-section-number">13</span> R environment set up</a></li>
<li><a class="" href="gprofiler2.html"><span class="header-section-number">14</span> gprofiler2</a></li>
<li><a class="" href="clusterprofiler-and-enrichplot.html"><span class="header-section-number">15</span> clusterProfiler and enrichplot</a></li>
<li><a class="" href="webgestaltr.html"><span class="header-section-number">16</span> WebGestaltR</a></li>
<li><a class="" href="novel-species-fea.html"><span class="header-section-number">17</span> Novel species FEA</a></li>
<li><a class="" href="workshop-wrap-up.html"><span class="header-section-number">18</span> Workshop wrap-up</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://monashbioinformaticsplatform.github.io/Functional_Enrichment_Online_Tools/">View book source <i class="fab fa-gitlab"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><span class="main-title"><em>Making sense of gene and proteins lists with functional enrichment analysis</em></span>
<div id="introduction" class="section level1" number="12">
<h1>
<span class="header-section-number">12</span> Introduction<a class="anchor" aria-label="anchor" href="#introduction"><i class="fas fa-link"></i></a>
</h1>
<p>For today’s session, we will be using the <a href="https://www.r-project.org/">R</a> statistical programming language to perform functional enrichmemt analysis using a range of dedicated R packages. To simplify our work, we will be using the <a href="https://posit.co/download/rstudio-desktop/">RStudio</a> integrated development environment.</p>
<p>RStudio provides a central place to write code, comments, run code, and view output. We will be using the workshop VMs today which run RStudio and have all the required R packages pre-installed.</p>
<div class="inline-figure"><img src="images/rstudio_logo.png" style="border: none; box-shadow: none; background: none;"></div>
<p>
 
</p>
<p><!-- insert blank line --></p>
<div id="day-2-overview" class="section level2" number="12.1">
<h2>
<span class="header-section-number">12.1</span> Day 2 overview<a class="anchor" aria-label="anchor" href="#day-2-overview"><i class="fas fa-link"></i></a>
</h2>
<ol style="list-style-type: decimal">
<li>R environment setup including VM login</li>
<li>ORA with <code>gprofiler2</code> including multi-query of up and down regulated genes</li>
<li>GSEA over KEGG database with <code>clusterProfiler</code> and multiple enrichment visualisations with <code>enrichplot</code>
</li>
<li>ORA and GSEA with <code>WebGestaltR</code>, interactive HTML reports, and exploration of term redundancy options</li>
<li>Novel species FEA with <code>clusterProfiler</code>, <code>WebGestaltR</code> and <code>STRING</code> (web)</li>
<li>End of workshop summary</li>
</ol>
<p>
 
</p>
<p><!-- insert blank line --></p>
</div>
<div id="r-fea-packages-for-fea" class="section level2" number="12.2">
<h2>
<span class="header-section-number">12.2</span> R FEA packages for FEA<a class="anchor" aria-label="anchor" href="#r-fea-packages-for-fea"><i class="fas fa-link"></i></a>
</h2>
<p>There are numerous R packages for FEA, each with their strenghts and limitations. It was difficult to restrict the chosen tools in order to fit within the time allocated for the workshop!</p>
<p>
 
</p>
<p><!-- insert blank line --></p>
<div class="inline-figure"><img src="images/r-tools-comparison-high-level.png" style="border: none; box-shadow: none; background: none; width: 100%;"></div>
<p>
 
</p>
<p><!-- insert blank line --></p>
<p>The following three R tools have been selected for today’s workshop:</p>
<ol style="list-style-type: decimal">
<li>
<code>gprofiler2</code> due to its ease of use, high number of supported species, and multiple database enrichments produced within a single run. Caveat: only ORA analysis</li>
<li>
<code>clusterProfiler2</code> due to its integrated database support, runs both GSEA and ORA, companion plotting tool <code>enrichplot</code> with diverse plot options, and novel species support</li>
<li>
<code>WebGestaltR</code> due to its ease of use, high number of suported databases and namespaces, runs both GSEA and ORA, interactive HTML reports and plots, and novel species support</li>
</ol>
</div>
<div id="day-2-approach" class="section level2" number="12.3">
<h2>
<span class="header-section-number">12.3</span> Day 2 approach<a class="anchor" aria-label="anchor" href="#day-2-approach"><i class="fas fa-link"></i></a>
</h2>
<p>For this part of the workshop, we will be doing a <strong>semi code-along</strong> approach, where you are not required to type any code (apart from a few basics in the RStudio familiarisation part next up) but we will be running code chunks together, talking about what each part of the code is doing, and viewing the output in real time.</p>
<p>All of the code required has been prepared into separate code files for each activity, and we will download these to the VMs in the next section.</p>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="resources.html"><span class="header-section-number">10</span> Resources</a></div>
<div class="next"><a href="r-environment-set-up.html"><span class="header-section-number">13</span> R environment set up</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#introduction"><span class="header-section-number">12</span> Introduction</a></li>
<li><a class="nav-link" href="#day-2-overview"><span class="header-section-number">12.1</span> Day 2 overview</a></li>
<li><a class="nav-link" href="#r-fea-packages-for-fea"><span class="header-section-number">12.2</span> R FEA packages for FEA</a></li>
<li><a class="nav-link" href="#day-2-approach"><span class="header-section-number">12.3</span> Day 2 approach</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://monashbioinformaticsplatform.github.io/Functional_Enrichment_Online_Tools//blob/master/09-day2-introduction.Rmd">View source <i class="fab fa-gitlab"></i></a></li>
          <li><a id="book-edit" href="https://monashbioinformaticsplatform.github.io/Functional_Enrichment_Online_Tools//edit/master/09-day2-introduction.Rmd">Edit this page <i class="fab fa-gitlab"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Functional Enrichment Workshop</strong>" was written by Australian Biocommons, Sydney Informatics Hub (USYD) and Monash Genomics and Bioinformatics Platform (MGBP). It was last built on <span>“October 14, 2025”</span> <br><img src="logos.png" width="300" alt="Logo"><br>.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
