<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 7 Online Tools | Functional Enrichment Workshop</title>
<meta name="author" content="Australian Biocommons, Sydney Informatics Hub (USYD) and Monash Genomics and Bioinformatics Platform (MGBP)">
<meta name="description" content="Making sense of gene and proteins lists with functional enrichment analysis  Functional enrichment analysis can be performed using various web-based tools, each of which is designed to meet...">
<meta name="generator" content="bookdown 0.40 with bs4_book()">
<meta property="og:title" content="Chapter 7 Online Tools | Functional Enrichment Workshop">
<meta property="og:type" content="book">
<meta property="og:description" content="Making sense of gene and proteins lists with functional enrichment analysis  Functional enrichment analysis can be performed using various web-based tools, each of which is designed to meet...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 7 Online Tools | Functional Enrichment Workshop">
<meta name="twitter:description" content="Making sense of gene and proteins lists with functional enrichment analysis  Functional enrichment analysis can be performed using various web-based tools, each of which is designed to meet...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.8.0/transition.js"></script><script src="libs/bs3compat-0.8.0/tabs.js"></script><script src="libs/bs3compat-0.8.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/kePrint-0.0.1/kePrint.js"></script><link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Functional Enrichment Workshop</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> Functional Enrichment Workshop</a></li>
<li><a class="" href="overview.html"><span class="header-section-number">2</span> Overview</a></li>
<li><a class="" href="recap.html"><span class="header-section-number">3</span> Recap</a></li>
<li><a class="" href="enrichment-statistics.html"><span class="header-section-number">4</span> Enrichment Statistics</a></li>
<li><a class="" href="example-analysis.html"><span class="header-section-number">5</span> Example Analysis</a></li>
<li><a class="" href="defining-the-genelist.html"><span class="header-section-number">6</span> Defining the genelist</a></li>
<li><a class="active" href="online-tools.html"><span class="header-section-number">7</span> Online Tools</a></li>
<li><a class="" href="reporting-results.html"><span class="header-section-number">8</span> Reporting results</a></li>
<li><a class="" href="resources.html"><span class="header-section-number">9</span> Resources</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://monashbioinformaticsplatform.github.io/Functional_Enrichment_Online_Tools/">View book source <i class="fab fa-gitlab"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><span class="main-title"><em>Making sense of gene and proteins lists with functional enrichment analysis</em></span>
<div id="online-tools" class="section level1" number="7">
<h1>
<span class="header-section-number">7</span> Online Tools<a class="anchor" aria-label="anchor" href="#online-tools"><i class="fas fa-link"></i></a>
</h1>
<p>Functional enrichment analysis can be performed using various web-based tools, each of which is designed to meet specific analytical needs. These tools often vary in the databases they use, their statistical approaches, and their capabilities to perform different types of analysis, such as Over-Representation Analysis (ORA) or Gene Set Enrichment Analysis (GSEA).</p>
<p>In this workshop, we will explore several popular tools for functional enrichment analysis, including gProfiler, STRING, Reactome, and MSigDB GSEA. Each tool offers unique features and insights, providing flexibility in selecting the right method for diverse datasets and research questions.</p>
<!-- ## FEA in [g:Profiler](https://biit.cs.ut.ee/gprofiler/gost) -->
<div id="fea-in-gprofiler" class="section level2" number="7.1">
<h2>
<span class="header-section-number">7.1</span> FEA in gProfiler <a href="https://biit.cs.ut.ee/gprofiler/gost" target="_blank"><img src="images/gp2-logo-new.png" alt="g:Profiler Logo" style="height:35px; vertical-align:middle;"></a><a class="anchor" aria-label="anchor" href="#fea-in-gprofiler"><i class="fas fa-link"></i></a>
</h2>
<p>gProfiler is known for its integration of numerous species and databases. It supports both ORA and GSEA, enabling users to assess Gene Ontology (GO), biological pathways, regulatory motifs and protein databases. With gProfile one can</p>
<div id="steps-to-perform-ora-in-gprofiler" class="section level3" number="7.1.1">
<h3>
<span class="header-section-number">7.1.1</span> Steps to perform ORA in g:Profiler:<a class="anchor" aria-label="anchor" href="#steps-to-perform-ora-in-gprofiler"><i class="fas fa-link"></i></a>
</h3>
<p><span style="color:orange;">- Prepare Input List:</span> Ensure your input is formatted as one gene per line or in a suitable format for g:Profiler.</p>
<p><span style="color:orange;">- Input Gene List:</span> Paste your prepared gene list directly into the input box on the g:Profiler web page or upload a file containing your list.</p>
<p><span style="color:orange;">- Select Organism:</span> Choose the appropriate organism from the <code>Organism</code> dropdown menu (e.g., <em>Homo sapiens</em> for human data).</p>
<p><span style="color:orange;">- Choose Statistical Domain Scope:</span> Under <code>Advanced options</code>, select your preferred statistical background from the <code>Statistical domain scope</code> menu.If you choose “Custom” background, provide your custom background list by pasting or uploading the relevant file.</p>
<p><span style="color:orange;">- Set Significance Threshold:</span> Select the desired significance threshold method, such as <em>g:SCS</em>, <em>Bonferroni</em>, or <em>Benjamini-Hochberg</em>.
- Specify the threshold value (e.g., 0.05, 0.1, etc.).</p>
<p><span style="color:orange;">- Select Functional Annotation Databases: </span> Navigate to the <code>Data sources</code> tab and choose one or more databases for analysis. Available options include:</p>
<ul>
<li>
<em>Gene Ontology (GO)</em>: Biological Process, Molecular Function, and Cellular Component.</li>
<li><em>KEGG Pathways</em></li>
<li><em>Reactome Pathways</em></li>
<li><em>WikiPathways</em></li>
<li><em>TRANSFAC</em></li>
<li><em>mirTarBase</em></li>
<li><em>Human Protein Atlas</em></li>
<li><em>CORUM</em></li>
<li><em>Human Phenotype Ontology (HP)</em></li>
</ul>
<p><span style="color:orange;">- Run Query:</span> Run the analysis and review the enriched terms, pathways, and visual outputs. Download the results as needed for further exploration.</p>
<div id="browse-the-gprofiler-results" class="section level4" number="7.1.1.1">
<h4>
<span class="header-section-number">7.1.1.1</span> Browse the gProfiler Results<a class="anchor" aria-label="anchor" href="#browse-the-gprofiler-results"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li><p><strong>Overview</strong>:
The analysis provided a comprehensive list of enriched terms across selected databases, highlighting significant GO. The results give a high-level summary of pathways or terms most relevant to the input data.</p></li>
<li><p><strong>Detailed Results</strong>:
The detailed results section includes a tabulated format with enriched terms, adjusted p-values, and relevant statistics. Each entry provides information such as the enrichment score, associated genes, and functional annotations, allowing for an in-depth understanding of biological significance.</p></li>
<li><p><strong>GO Context</strong>:
The Gene Ontology (GO) context is divided into three main categories: Biological Process (BP), Molecular Function (MF), and Cellular Component (CC). The analysis identifies which GO terms are significantly enriched, offering insights into the broader biological implications of the gene set. This helps in pinpointing processes such as cellular responses, metabolic pathways, and molecular interactions.</p></li>
<li><p><strong>Query Info</strong>:
This section includes specifics about the input data, including the total number of queried genes and any identifiers not recognised or mapped. It also details the statistical background used, the chosen organism, and other analysis settings, ensuring transparency and reproducibility of the results.</p></li>
</ul>
</div>
<div id="section-3" class="section level4 unnumbered">
<h4 class="unnumbered"><a class="anchor" aria-label="anchor" href="#section-3"><i class="fas fa-link"></i></a></h4>
</div>
<div id="different-backgrounds" class="section level4" number="7.1.1.2">
<h4>
<span class="header-section-number">7.1.1.2</span> Different Backgrounds<a class="anchor" aria-label="anchor" href="#different-backgrounds"><i class="fas fa-link"></i></a>
</h4>
</div>
<div id="challenge-how-different-backgrounds-impact-the-output" class="section level4 unnumbered challenge">
<h4>
<strong>Challenge:</strong> How different backgrounds impact the output?<a class="anchor" aria-label="anchor" href="#challenge-how-different-backgrounds-impact-the-output"><i class="fas fa-link"></i></a>
</h4>
<p>Use ‘All known genes’ in one analysis and ‘Custom’ background in another. Download the results by clicking on CSV button. Browse the results in the spreadsheets and find out the difference between two.</p>
</div>
<div id="questions-1" class="section level4 unnumbered rationale">
<h4>
<strong>Questions </strong><a class="anchor" aria-label="anchor" href="#questions-1"><i class="fas fa-link"></i></a>
</h4>
<p>Which background would you use in your analysis?</p>
<p>How is multi-query support implemented in gProfiler?</p>
<p>How can one perform Under Representation Analysis in gProfiler?</p>
</div>
</div>
<div id="steps-to-perform-gsea-in-gprofiler" class="section level3" number="7.1.2">
<h3>
<span class="header-section-number">7.1.2</span> Steps to perform GSEA in g:Profiler:<a class="anchor" aria-label="anchor" href="#steps-to-perform-gsea-in-gprofiler"><i class="fas fa-link"></i></a>
</h3>
<p><span style="color:orange;">- Prepare Your Pre-ranked List:</span> Steps to provide a ranked gene list are given <a href="degust.html">here</a>.</p>
<p><span style="color:orange;">- Input Gene List:</span> Paste your prepared gene list directly into the input box on the g:Profiler web page or upload a file containing your list.</p>
<p><span style="color:orange;">- Select Organism:</span> Same as above.</p>
<p><span style="color:orange;">- Select Ordered query:</span> The “Ordered query” option in g:Profiler is designed to work with pre-ranked gene lists.</p>
<p><span style="color:orange;">- Set Significance Threshold:</span> Same as above.</p>
<p><span style="color:orange;">- Provide a Custom GMT:</span> This GMT file can be downloaded from <a href="https://www.gsea-msigdb.org/gsea/msigdb/index.jsp">MSigDB</a>.</p>
<p><span style="color:orange;">- Run Query:</span> Same as above.</p>
<div id="challenge-gsea-with-gprofiler" class="section level4 unnumbered challenge">
<h4>
<strong>Challenge:</strong> GSEA with gProfiler<a class="anchor" aria-label="anchor" href="#challenge-gsea-with-gprofiler"><i class="fas fa-link"></i></a>
</h4>
<p>Download the Hallmark gene sets (<a href="https://www.gsea-msigdb.org/">h.all.v2024.1.Hs.symbols.gmt</a>) from MSigDB and use it as Custom GMT.</p>
<!-- Full path to h.all.v2024.1.Hs.symbols.gmt -->
<!-- [h.all.v2024.1.Hs.symbols.gmt](https://www.gsea-msigdb.org/gsea/msigdb/download_file.jsp?filePath=/msigdb/release/2024.1.Hs/h.all.v2024.1.Hs.symbols.gmt) -->
</div>
<div id="question" class="section level4 unnumbered rationale">
<h4>
<strong>Question </strong><a class="anchor" aria-label="anchor" href="#question"><i class="fas fa-link"></i></a>
</h4>
<p>How can one use multi-GMT as custom background?</p>
<!-- ## [STRING](https://string-db.org) -->
</div>
</div>
</div>
<div id="fea-in-string" class="section level2" number="7.2">
<h2>
<span class="header-section-number">7.2</span> FEA in STRING <a href="https://string-db.org" target="_blank"><img src="images/string-logo.png" alt="STRING Logo" style="height:35px; vertical-align:middle;"></a><a class="anchor" aria-label="anchor" href="#fea-in-string"><i class="fas fa-link"></i></a>
</h2>
<p>STRING (Search Tool for the Retrieval of Interacting Genes/Proteins) is a resource for exploring protein-protein interaction (PPI) networks. It combines experimental data, predictions, and curated information to build networks that highlight functional relationships, helping to reveal shared pathways or biological processes within gene or protein lists.</p>
<div id="steps-to-perform-ora-in-string" class="section level3" number="7.2.1">
<h3>
<span class="header-section-number">7.2.1</span> Steps to Perform ORA in STRING:<a class="anchor" aria-label="anchor" href="#steps-to-perform-ora-in-string"><i class="fas fa-link"></i></a>
</h3>
<p><span style="color:orange;">- Select Multiple proteins tab.</span></p>
<p><span style="color:orange;">- Input Gene List:</span> Paste your prepared gene list directly into the input box on the STRING web page or upload a file containing your list.</p>
<p><span style="color:orange;">- Select Organism:</span> Choose the appropriate organism from the <code>Organisms</code> dropdown menu (e.g., <em>Homo sapiens</em> for human data). STRING would auto-detect the organism if ENSEMBL IDs provided.</p>
<p><span style="color:orange;">- Modify Settings:</span> Under <code>Advanced Settings</code>, you can modify <code>Required score</code> from low (0.15) to highest (0.9) confidence. Similarly <code>FDR stringency</code> and <code>Network type</code> can be selected.</p>
<p>NOTE: In cases where long list of features is provided, STRING may chnage some of its settings so that:</p>
<ul>
<li>the nodes will have a simplified (not 3D) design</li>
<li>previews of protein structures are not shown</li>
<li>the network edges show interaction confidence only</li>
</ul>
</div>
<div id="browse-the-string-ora-results" class="section level3" number="7.2.2">
<h3>
<span class="header-section-number">7.2.2</span> Browse the STRING ORA Results<a class="anchor" aria-label="anchor" href="#browse-the-string-ora-results"><i class="fas fa-link"></i></a>
</h3>
<p>STRING generates multiple tabs as output, shown here:</p>
<!-- ![](images/string-results-tabs.png){ width=100% } -->
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-17"></span>
<img src="images/string-results-tabs.png" alt="Results tabs in STRING" width="100%"><p class="caption">
Figure 7.1: Results tabs in STRING
</p>
</div>
<div id="viewers" class="section level4" number="7.2.2.1">
<h4>
<span class="header-section-number">7.2.2.1</span> Viewers<a class="anchor" aria-label="anchor" href="#viewers"><i class="fas fa-link"></i></a>
</h4>
<p>Under the <code>Viewers</code> tab, various visualisation layouts are available, with the Network option being the most notable and widely used.</p>
</div>
<div id="legend" class="section level4" number="7.2.2.2">
<h4>
<span class="header-section-number">7.2.2.2</span> Legend<a class="anchor" aria-label="anchor" href="#legend"><i class="fas fa-link"></i></a>
</h4>
<p>The <code>Legend</code> tab offers a guide to the colors of nodes and edges, along with annotations for each individual query in the input list.</p>
<!-- ![Nodes and edges colour-coded](images/string-legend.png){ width=100% } -->
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-18"></span>
<img src="images/string-legend.png" alt="Nodes and edges colour-coded" width="100%"><p class="caption">
Figure 7.2: Nodes and edges colour-coded
</p>
</div>
</div>
<div id="settings" class="section level4" number="7.2.2.3">
<h4>
<span class="header-section-number">7.2.2.3</span> Settings<a class="anchor" aria-label="anchor" href="#settings"><i class="fas fa-link"></i></a>
</h4>
<p>In the <code>Settings</code> tab of the STRING results, users have the flexibility to adjust existing settings and apply new filters to customise their data view and analysis. This tab you to switch between network types, strengths, data sources, intertaction scores and more.</p>
</div>
<div id="analysis" class="section level4" number="7.2.2.4">
<h4>
<span class="header-section-number">7.2.2.4</span> Analysis<a class="anchor" aria-label="anchor" href="#analysis"><i class="fas fa-link"></i></a>
</h4>
<p>One of the most essential tabs is the <code>Analysis</code> tab, which offers comprehensive functional enrichment analysis from a range of databases. These include Gene Ontology (GO) for biological processes, molecular functions, and cellular components; Pathway enrichment from sources such as KEGG, Reactome, and WikiPathways; and other significant data sources such as Human Phenotype annotations and UniProt for protein function and structure.</p>
<p>Columns of the STRING enrichment table are explained as following:</p>
<!-- ![Enrichment Table Columns](images/Enrichment-Table-Columns.png){ width=40% } -->
<p><span style="color:orange;">- Count In Network:</span>
The first number indicates how many proteins in your network are annotated with a particular term. The second number indicates how many proteins in total (in your network and in the background) have this term assigned. You can click on the numbers to see the network view of the gene sets behind them.</p>
<p><span style="color:orange;">- Strength:</span>
Log10(observed / expected). This measure describes how large the enrichment effect is. It’s the ratio between i) the number of proteins in your network that are annotated with a term and ii) the number of proteins that we expect to be annotated with this term in a random network of the same size.</p>
<p><span style="color:orange;">- Signal:</span>
The signal is defined as a weighted harmonic mean between the observed/expected ratio and -log(FDR). FDR tends to emphasise larger terms due to their potential for achieving lower p-values, while the observed/expected ratio highlights smaller terms, which have a high foreground to background ratio but cannot achieve low FDR values due to their size. The signal measure seeks to balance both metrics for a more intuittive ordering of enriched terms.</p>
<p><span style="color:orange;">- False Discovery Rate:</span>
This measure describes how significant the enrichment is. Shown are p-values corrected for multiple testing within each category using the Benjamini–Hochberg procedure.</p>
<p>STRING visualises terms within each category using a bubble plot, effectively showcasing the significance and size of enriched terms. Additionally, it renders groups of related terms based on a user-defined similarity level, allowing users to identify clusters of functionally related terms within the data. This helps in interpreting complex enrichment results and highlighting key biological processes or pathways that are closely associated.</p>
<!-- ![Functional enrichment visualisation with STRING](images/string-enrichment_KEGG_sim0.7_graph_plus.png){ width=100% } -->
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-19"></span>
<img src="images/string-enrichment_KEGG_sim0.7_graph_plus.png" alt="Functional enrichment visualisation with STRING" width="100%"><p class="caption">
Figure 7.3: Functional enrichment visualisation with STRING
</p>
</div>
Towards the the bottom of the <code>Analysis</code> page, one can change the background including adding one of their own.
<!-- ![Statistical background](images/string-statistical-background.png){ width=100% } -->
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-20"></span>
<img src="images/string-statistical-background.png" alt="Statistical background" width="100%"><p class="caption">
Figure 7.4: Statistical background
</p>
</div>
<p>Finally the enriched terms can be downloaded at the end of the <code>Analysis</code> page, either individually per category or all enriched terms together.</p>
</div>
<div id="exports" class="section level4" number="7.2.2.5">
<h4>
<span class="header-section-number">7.2.2.5</span> Exports<a class="anchor" aria-label="anchor" href="#exports"><i class="fas fa-link"></i></a>
</h4>
<p>The network data can be exported with the <code>Exports</code> tab. Also Network data can be directly sent to <a href="https://cytoscape.org/">Cytoscape</a> <img src="images/network-to-Cytoscape.png"> for further networking. It is expected to have Cytoscape installed before exporting to it.</p>
</div>
<div id="clusters" class="section level4" number="7.2.2.6">
<h4>
<span class="header-section-number">7.2.2.6</span> Clusters<a class="anchor" aria-label="anchor" href="#clusters"><i class="fas fa-link"></i></a>
</h4>
<p>The <code>Clusters</code> tab essentially provides three different types of clustering algorithms:</p>
<ul>
<li><p>k-means clustering: Initialise <em>k</em> centroids randomly, assigns each data point to the nearest centroid, recompute the centroids as the mean of all points in a cluster until until centroids do not change significantly.</p></li>
<li><p>MCL clustering (Markov clustering): is a graph-based algorithm that uses flow simulation to detect clusters in a network by modeling random walks.</p></li>
<li><p>DBSCAN clustering: is a density-based algorithm that groups points closely packed together while marking points in low-density regions as outliers or noise</p></li>
</ul>
<!-- ![](images/string-clusters.png) --><div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-21"></span>
<img src="images/string-clusters.png" alt="Network clustering in STRING" width="100%"><p class="caption">
Figure 7.5: Network clustering in STRING
</p>
</div>
<p>Clusters can be downloaded in <code>.tsv</code> format.</p>
</div>
<div id="question-1" class="section level4 unnumbered rationale">
<h4>
<strong>Question</strong><a class="anchor" aria-label="anchor" href="#question-1"><i class="fas fa-link"></i></a>
</h4>
<p>What was the overlap in enrichment terms between gProfiler and STRING at FDR ≤ 0.05?</p>
</div>
</div>
<div id="steps-to-perform-gsea-in-string" class="section level3" number="7.2.3">
<h3>
<span class="header-section-number">7.2.3</span> Steps to Perform GSEA in STRING:<a class="anchor" aria-label="anchor" href="#steps-to-perform-gsea-in-string"><i class="fas fa-link"></i></a>
</h3>
<p><span style="color:orange;">- Selcet Proteins with Values/Ranks.</span></p>
<p><span style="color:orange;">- Input Gene List:</span> Paste your gene list with a meaningful value for ranking (fold-change, log-pvalue, abundance, …) directly into the input box on the STRING web page or upload a file containing your list of features and their corresponding values.</p>
<p><span style="color:orange;">- Select Organism:</span> Same as above.</p>
<p><span style="color:orange;">- Advanced Setting:</span> FDR stringency and the initial sort order can be set up in advance and hit the Search.</p>
</div>
<div id="browse-the-string-gsea-results" class="section level3" number="7.2.4">
<h3>
<span class="header-section-number">7.2.4</span> Browse the STRING GSEA Results<a class="anchor" aria-label="anchor" href="#browse-the-string-gsea-results"><i class="fas fa-link"></i></a>
</h3>
<p>The output differs from ORA. For each gene set, the results include the enrichment score, its direction within the ranked list, the number of overlapping features with the gene set, and the associated FDR.</p>
<p>When a user selects a gene set from the enriched table,</p>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-22"></span>
<img src="images/string-gsea-wikiPathways.png" alt="An example table of WikiPathway gene sets" width="100%"><p class="caption">
Figure 7.6: An example table of WikiPathway gene sets
</p>
</div>
<p>the associated genes are displayed within the ranking list. A table showing these genes along with their original ranking values is also provided.</p>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-23"></span>
<img src="images/string-gsea-ranking-n-table.png" alt="List of genes in the term (WP197) and their positions on the ranked list" width="286"><p class="caption">
Figure 7.7: List of genes in the term (WP197) and their positions on the ranked list
</p>
</div>
<p>Additionally, the locations of the corresponding proteins are highlighted in the proteome network:</p>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-24"></span>
<img src="images/string-gsea-proteome-network.png" alt="Proteome network" width="292"><p class="caption">
Figure 7.8: Proteome network
</p>
</div>
<p>A Functional enrichment visualisation (similar to that of ORA) is provided at below the enriched tables.</p>
<p>Modify <code>Enrichment display settings</code> tab before downloading the enriched tables. It is recommended to merge terms with a certain level of similarity to reduce redundancy, especially if there are many overlapping terms.</p>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-25"></span>
<img src="images/string-enrichement-display-settings.png" alt="Enrichment display settings" width="277"><p class="caption">
Figure 7.9: Enrichment display settings
</p>
</div>
<p>Here is an example output of <a href="https://version-12-0.string-db.org/cgi/globalenrichment?networkId=bKhJ4fXp6sna">GSEA on STRING</a> from a previous run (the link will expire in future).</p>
<!-- ## FEA in [GenePattern](https://www.genepattern.org/#gsc.tab=0) -->
</div>
</div>
<div id="fea-in-genepattern" class="section level2" number="7.3">
<h2>
<span class="header-section-number">7.3</span> FEA in GenePattern <a href="https://www.genepattern.org/#gsc.tab=0" target="_blank"><img src="images/GenePattern-logo.png" alt="GenePattern Logo" style="height:35px; vertical-align:middle;"></a><a class="anchor" aria-label="anchor" href="#fea-in-genepattern"><i class="fas fa-link"></i></a>
</h2>
<p>GenePattern, an online platform developed by the Broad Institute, offers a suite of tools for analyzing and visualizing genomic data, making bioinformatics accessible to researchers through a user-friendly, no-programming interface. Among its supported tools is Gene Set Enrichment Analysis (GSEA), which implements <a href="https://www.gsea-msigdb.org/gsea/index.jsp">MSigDB GSEA</a> analysis for identifying enriched gene sets in genomic data.</p>
<p>MSigDB (Molecular Signatures Database) is a collection of gene sets for Gene Set Enrichment Analysis, representing pathways and gene signatures linked to biological states or diseases. It helps identify enriched gene sets, aiding the analysis of gene expression changes and key pathways in experimental data.</p>
<div id="steps-to-locate-gsea-module-in-genepattern" class="section level3" number="7.3.1">
<h3>
<span class="header-section-number">7.3.1</span> Steps to Locate GSEA Module in GenePattern:<a class="anchor" aria-label="anchor" href="#steps-to-locate-gsea-module-in-genepattern"><i class="fas fa-link"></i></a>
</h3>
<ul>
<li>Click on the Run button and then the Public Server</li>
</ul>
<!-- ![](images/GenePattern-Run.png) --><div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-26"></span>
<img src="images/GenePattern-Run.png" alt="Navigate to Public Server" width="286"><p class="caption">
Figure 7.10: Navigate to Public Server
</p>
</div>
<ul>
<li><p>Sign in to GenePattern or Enter as Guest</p></li>
<li><p>Under <code>Modules</code> tab hit <code>Browse Modules</code></p></li>
<li><p>Find gsea in the Browse Modules by Category page and hit GSEA</p></li>
</ul>
<!-- ![](images/Browse_Modules_gsea.png){ width=100% } --><div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-27"></span>
<img src="images/Browse_Modules_gsea.png" alt="Browse GSEA module in GenePattern" width="100%"><p class="caption">
Figure 7.11: Browse GSEA module in GenePattern
</p>
</div>
</div>
<div id="steps-to-perform-gsea" class="section level3" number="7.3.2">
<h3>
<span class="header-section-number">7.3.2</span> Steps to Perform GSEA:<a class="anchor" aria-label="anchor" href="#steps-to-perform-gsea"><i class="fas fa-link"></i></a>
</h3>
<!-- https://cloud.genepattern.org/gp/pages/index.jsf?jobid=613752&openVisualizers=true&openNewWindow=false -->
<ol style="list-style-type: decimal">
<li>Basic Parameters</li>
</ol>
<p>- Create both <code>.gct</code> and <code>.cls</code> files following <a href="degust.html">this scrit in R</a></p>
<p>- Load the <code>.gct</code> input file in the <code>expression dataset</code> tab and <code>.cls</code> file in the <code>phenotype labels</code> tab</p>
<p>- Select a <code>.gmt</code> file (Gene Matrix Transposed) from the <code>gene sets database</code> tab</p>
<p>- Set permutation under <code>number of permutations</code> tab</p>
<p>- Type of the permuattaion to be set under <code>permutation type</code> tab</p>
<p>- Select an appropriate DNA Chip annotation file from <code>chip platform file</code> tab</p>
<p>- Name the output file in <code>output file name</code> tab</p>
<ol start="2" style="list-style-type: decimal">
<li>Advanced Parameters</li>
</ol>
<p>- Scoring Scheme:</p>
<ul>
<li><p>K-S: The score increment is the same for all genes in <em>S</em> regardless of their ranking or correlation strength.</p></li>
<li><p>WEighted: the score increment for each gene in <em>S</em> is weighted by its correlation with the phenotype, typically the absolute value of the correlation or ranking metric.</p></li>
</ul>
<p>- Metric for ranking genes: Ranking metric of interest can be chosen from drop down menu. A detailed description od the metrics is given on <a href="https://docs.gsea-msigdb.org/#GSEA/GSEA_User_Guide/#metrics-for-ranking-genes">GSEA-MSigDB Documentation</a>.</p>
<ul>
<li><p>Categorical Phenotypes: Signal-to-Noise Ratio, t-Test, Ratio of Classes, Log2 Ratio of Classes</p></li>
<li><p>Continuous Phenotypes: Pearson Correlation, Spearman Correlation</p></li>
</ul>
<p>- Minimum and Maximum size of gene sets can be set using <code>max gene set size</code> and <code>min gene set size</code> tabs
<!-- <GeneSetName> <Description> <Gene1> <Gene2> <Gene3> ... --></p>
<div id="browse-the-gsea-results" class="section level4" number="7.3.2.1">
<h4>
<span class="header-section-number">7.3.2.1</span> Browse the GSEA results<a class="anchor" aria-label="anchor" href="#browse-the-gsea-results"><i class="fas fa-link"></i></a>
</h4>
<p>Once the job has been queued and successfully run, the output will be listd on the left panel under <code>Jobs</code> tab:</p>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-28"></span>
<img src="images/GenePattern-Jobs.png" alt="Job status in GenePattern" width="172"><p class="caption">
Figure 7.12: Job status in GenePattern
</p>
</div>
<p>Of the most important files is the <code>.zip</code> file that was earlier specified under <code>output file name</code> tab in Basic parameters section which includes all the results. The results can also be navigated using the single files listed under the job id.</p>
<p>For Pezzini experiment, two <code>html</code> files generated for each of up- and down-regulated gene sets, something like:</p>
<ul>
<li><p>gsea_report_for_Diff_1731388275794.html</p></li>
<li><p>gsea_report_for_Nodiff_1731388275794.html</p></li>
</ul>
<p>The tabulated versions of the results are given in <code>.tsv</code> format:</p>
<ul>
<li><p>gsea_report_for_Diff_1731388275794.tsv</p></li>
<li><p>gsea_report_for_Nodiff_1731388275794.tsv</p></li>
</ul>
<!-- ```{css, echo=FALSE} --><!-- table { --><!--   width: 75%;                   --><!--   margin: auto;                 --><!--   border-collapse: collapse;    --><!-- } --><!-- th, td { --><!--   padding: 5px;                 --><!--   text-align: left; --><!--   border: 1px solid #ddd;       --><!-- } --><!-- ``` --><p>The GSEA result tables have the following header and below is given details of one gene set:</p>
<div class="inline-table"><table class="table table-sm">
<caption>
<span id="tab:unnamed-chunk-29">Table 7.1: </span>Summary of GSEA Results for REACTOME_FRS_MEDIATED_FGFR2_SIGNALING Gene Set</caption>
<colgroup>
<col width="41%">
<col width="58%">
</colgroup>
<thead><tr class="header">
<th align="left">Parameter</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">GS (follow link to MSigDB)</td>
<td align="left"><a href="https://www.gsea-msigdb.org/gsea/msigdb/human/geneset/REACTOME_FRS_MEDIATED_FGFR2_SIGNALING">REACTOME_FRS_MEDIATED_FGFR2_SIGNALING</a></td>
</tr>
<tr class="even">
<td align="left">GS DETAILS</td>
<td align="left">Details …</td>
</tr>
<tr class="odd">
<td align="left">SIZE</td>
<td align="left">16</td>
</tr>
<tr class="even">
<td align="left">ES</td>
<td align="left">0.83905387</td>
</tr>
<tr class="odd">
<td align="left">NES</td>
<td align="left">1.7128055</td>
</tr>
<tr class="even">
<td align="left">NOM p-val</td>
<td align="left">0</td>
</tr>
<tr class="odd">
<td align="left">FDR q-val</td>
<td align="left">0.03902518</td>
</tr>
<tr class="even">
<td align="left">FWER p-val</td>
<td align="left">0.648</td>
</tr>
<tr class="odd">
<td align="left">RANK AT MAX</td>
<td align="left">995</td>
</tr>
<tr class="even">
<td align="left">LEADING EDGE</td>
<td align="left">tags=38%, list=7%, signal=40%</td>
</tr>
</tbody>
</table></div>
<!-- HALLMARK_CHOLESTEROL_HOMEOSTASIS --><!-- 68 0.59    1.49    0.012   0.059   0.201   4015    tags=60%, list=28%, signal=83% --><p>The leading edge column has three values:</p>
<ul>
<li>tags: 38% of the genes in the gene set are key to the enrichment result.</li>
<li>list: These genes make up 7% of the total gene list being analyzed.</li>
<li>signal: They contribute 40% of the enrichment signal, highlighting their importance in driving the association between this gene set and the biological phenotype being studied.</li>
</ul>
</div>
<div id="challenge-how-do-different-ranking-metrics-impact-the-output" class="section level4 unnumbered challenge">
<h4>
<strong>Challenge:</strong> How do different ranking metrics impact the output?<a class="anchor" aria-label="anchor" href="#challenge-how-do-different-ranking-metrics-impact-the-output"><i class="fas fa-link"></i></a>
</h4>
<p>Run GSEA analysis using Hallmark gene sets with two metrics (tTest and Ratio_of_Classes). What are the upregulated terms (FDR &lt; 0.1) in the <code>Diff</code> class, based on the t-test and Ratio of Classes metrics?</p>
</div>
<div id="question-2" class="section level4 unnumbered rationale">
<h4>
<strong>Question </strong><a class="anchor" aria-label="anchor" href="#question-2"><i class="fas fa-link"></i></a>
</h4>
<p>Why might the HALLMARK_CHOLESTEROL_HOMEOSTASIS gene set be upregulated specifically in the differentiation condition of SH-SY5Y cells in <a href="https://pubmed.ncbi.nlm.nih.gov/27422411/">Pezzini, et al 2016</a> experiment?</p>
<details><summary>
Show
</summary><ul>
<li>Relevance: Cholesterol is essential for neuronal function and membrane fluidity, particularly in processes like axonal growth and synapse formation. Neurons have a high demand for cholesterol, especially during differentiation when they extend axons and dendrites.</li>
<li>Possible Insight: Upregulation of genes in this set could signify that differentiating cells are actively producing or transporting cholesterol to support membrane synthesis and cellular remodeling required for mature neuronal structures.</li>
</ul></details>
</div>
<div id="section-4" class="section level4 unnumbered">
<h4 class="unnumbered"><a class="anchor" aria-label="anchor" href="#section-4"><i class="fas fa-link"></i></a></h4>
<!-- ## FEA in [Reactome](https://reactome.org/) -->
</div>
</div>
</div>
<div id="fea-in-reactome" class="section level2" number="7.4">
<h2>
<span class="header-section-number">7.4</span> FEA in Reactome <a href="https://reactome.org/" target="_blank"><img src="images/reactome-logo.png" alt="Reactome Logo" style="height:35px; vertical-align:middle;"></a><a class="anchor" aria-label="anchor" href="#fea-in-reactome"><i class="fas fa-link"></i></a>
</h2>
<p>Reactome is an open-source database of curated biological pathways across species, offering pathway maps and enrichment tools to analyse gene lists in a pathway-focused context. It’s ideal for visualising data within established biochemical and cellular processes.</p>
<div id="steps-to-perform-ora-in-reactome" class="section level3" number="7.4.1">
<h3>
<span class="header-section-number">7.4.1</span> Steps to perform ORA in Reactome:<a class="anchor" aria-label="anchor" href="#steps-to-perform-ora-in-reactome"><i class="fas fa-link"></i></a>
</h3>
<ul>
<li>Hit the <code>Analysis Tools</code> tab</li>
</ul>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-30"></span>
<img src="images/reactome-tabs.png" alt="Analysis in Reactome" width="100%"><p class="caption">
Figure 7.13: Analysis in Reactome
</p>
</div>
<ul>
<li>Choose <code>Analyse gene list</code> from the left panel</li>
</ul>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-31"></span>
<img src="images/reactome-analysis-tools.png" alt="Analysis Tools in Reactome" width="20%"><p class="caption">
Figure 7.14: Analysis Tools in Reactome
</p>
</div>
<ul>
<li><p>Upload list of features on the box or choose a file, hit continue</p></li>
<li>
<p>Select prefered options:</p>
<p>- Project to Human: This option will convert identifiers from non-human species into human equivalents, allowing you to analyse data across species.</p>
<p>- Include interactors: This option integrates interactors from IntAct, a protein interaction database. Including interactors broadens the background network, potentially offering deeper insights.</p>
</li>
<li><p>Hit Analyse!</p></li>
</ul>
</div>
<div id="steps-to-perform-gsa-in-reactome" class="section level3" number="7.4.2">
<h3>
<span class="header-section-number">7.4.2</span> Steps to perform GSA in Reactome:<a class="anchor" aria-label="anchor" href="#steps-to-perform-gsa-in-reactome"><i class="fas fa-link"></i></a>
</h3>
<ul>
<li>If <code>Analyse gene expression</code> was chosen instead, Reactome offers the following gene set analysis:</li>
</ul>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-32"></span>
<img src="images/reactome-gsea-methods.png" alt="Ractome GSA " width="100%"><p class="caption">
Figure 7.15: Ractome GSA
</p>
</div>
<ul>
<li><p>Lets try CAMERA as it represents the <code>camera()</code> function of <code>limma</code> package in <code>R</code> for a gene set analysis.</p></li>
<li><p>Choose TMM normalisation to ensure consistency with the input data used in other tools within our workshop.</p></li>
<li><p>Select data type and provide input data</p></li>
</ul>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-33"></span>
<img src="images/reactome-gsea-select-data.png" alt="Ractome GSA - data type" width="100%"><p class="caption">
Figure 7.16: Ractome GSA - data type
</p>
</div>
<ul>
<li>Annotate columns by adding extra info as follows:</li>
</ul>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-34"></span>
<img src="images/reactome-gsea-add-column.png" alt="Pathway diagram " width="100%"><p class="caption">
Figure 7.17: Pathway diagram
</p>
</div>
<ul>
<li><p>Save dataset and Continue</p></li>
<li><p>You can now browse the results</p></li>
</ul>
</div>
<div id="browse-the-reactome-results" class="section level3" number="7.4.3">
<h3>
<span class="header-section-number">7.4.3</span> Browse the Reactome results<a class="anchor" aria-label="anchor" href="#browse-the-reactome-results"><i class="fas fa-link"></i></a>
</h3>
<p>Results can be interactively browsed using the reactome pathway or voronoi visualisation modes: <a target="_blank"><img src="images/reactome-modes.png" alt="Reactome Modes" style="height:35px; vertical-align:middle;"></a></p>
<p>User can explore the pathway names listed in the table within the <code>Analysis</code> tab and they are displayed as popups on the pathway diagrams.</p>
<p>One can also select a pathway of interest by navigating through the left panel or by simply searching for the term in the search box.</p>
<p>The enriched table can be downloaded as shown below:</p>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-35"></span>
<img src="images/reactome-download-table.png" alt="Table of ORA with Reactome" width="100%"><p class="caption">
Figure 7.18: Table of ORA with Reactome
</p>
</div>
<p>The diagram can be downloaded using this icons: <a target="_blank"><img src="images/reactome-download-diagram.png" alt="Download Diagram" style="height:35px; vertical-align:middle;"></a></p>
<p>Here is a sample pathway diagram from Reactome GSA.</p>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-36"></span>
<img src="images/reactome-PathwaysOverview-expression-data.png" alt="Pathway diagram - GSA" width="100%"><p class="caption">
Figure 7.19: Pathway diagram - GSA
</p>
</div>
<p>In case <code>ssGSEA</code> was selected, an overall output would look like below:</p>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-37"></span>
<img src="images/reactome-ssGSEA.png" alt="Expression of top 30 pathways with ssGSEA" width="70%"><p class="caption">
Figure 7.20: Expression of top 30 pathways with ssGSEA
</p>
</div>
<div id="section-5" class="section level4 unnumbered">
<h4 class="unnumbered"><a class="anchor" aria-label="anchor" href="#section-5"><i class="fas fa-link"></i></a></h4>
</div>
<div id="question-3" class="section level4 unnumbered rationale">
<h4>
<strong>Question </strong><a class="anchor" aria-label="anchor" href="#question-3"><i class="fas fa-link"></i></a>
</h4>
<p>When running FEA in Reactome, how do you prefer the analysis methods?</p>
<p>- PADOG (Pathway Analysis with Down-weighting of Overlapping Genes)</p>
<p>- CAMERA (Correlation Adjusted Mean Rank)</p>
<p>- ssGSEA (Single Sample Gene Set Enrichment Analysis)</p>
</div>
<div id="section-6" class="section level4 unnumbered">
<h4 class="unnumbered"><a class="anchor" aria-label="anchor" href="#section-6"><i class="fas fa-link"></i></a></h4>
</div>
</div>
</div>
<div id="uncertainties-of-a-functional-enrichment-analsysis" class="section level2" number="7.5">
<h2>
<span class="header-section-number">7.5</span> Uncertainties of a functional enrichment analsysis<a class="anchor" aria-label="anchor" href="#uncertainties-of-a-functional-enrichment-analsysis"><i class="fas fa-link"></i></a>
</h2>
<p>This section provides a summary of the paper by <a href="https://wires.onlinelibrary.wiley.com/doi/full/10.1002/wics.1643">Wünsch et al. (2023)</a>, which explores uncertainties inherent in functional enrichment analysis. The study critically examines the sources of variability and challenges in this analytical approach, offering insights into improving its reliability and robustness.</p>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-38"></span>
<img src="images/Wunsch_et_al_2023.png" alt="From RNA sequencing measurements to the final results: A practical guide to navigating the choices and uncertainties of gene set analysis" width="100%"><p class="caption">
Figure 7.21: From RNA sequencing measurements to the final results: A practical guide to navigating the choices and uncertainties of gene set analysis
</p>
</div>
<div id="types-of-fea" class="section level3" number="7.5.1">
<h3>
<span class="header-section-number">7.5.1</span> <span style="color:red;">Types of FEA</span><a class="anchor" aria-label="anchor" href="#types-of-fea"><i class="fas fa-link"></i></a>
</h3>
<p>Functional enrichment analysis (FEA) typically involves one of over representation analysis (ORA), gene set enrichment analysis (GSEA) also known as functional class scoring (FCS), and Pathway Topology (PT).</p>
<ol style="list-style-type: decimal">
<li><strong>ORA</strong></li>
</ol>
<p>- ORA methods are the least complex among the three approaches of FEA.</p>
<p>- ORA methods requires a list of differentially expressed genes that are already analysed in differential expression analysis.</p>
<p>- The background population, the universe, can be a more general set of gene like those in human genome or more specific from thos observed in an experiment.</p>
<p>- A contingency table is created and the null distribution is modeled using the hypergeometric distribution.</p>
<ol start="2" style="list-style-type: decimal">
<li><strong>FCS</strong></li>
</ol>
<p>- FCS methods aim to aggregate the values of the gene-level statistics (ranks) into gene set-level statistic (enrichment score, ES).</p>
<p>- FCS can be classified as one of FCS I, those that take the expression data as input or FCS II that take a pre-ranked list of genes as input. With the latter, the information of the conditions (phenotypes) of the samples is lost, as such phenotype permutation cannot be performed leaving the choice of null hypothesis to gene set permutation.</p>
<ol start="3" style="list-style-type: decimal">
<li><strong>PT</strong></li>
</ol>
<p>- PT additionally models interactions between the genes. This approach generally scores considerably lower in terms of popularity in the reference database.</p>
</div>
<div id="considerations" class="section level3" number="7.5.2">
<h3>
<span class="header-section-number">7.5.2</span> <span style="color:red;">Considerations</span><a class="anchor" aria-label="anchor" href="#considerations"><i class="fas fa-link"></i></a>
</h3>
<p>- Pre-filter expression data: Exclude lowly expressed genes to improve statistical power.</p>
<p>- Handle gene IDs carefully: Convert gene IDs to the required format and remove any duplicates.</p>
<p>- Normalise expression data: Address sample-specific biases to enable fair comparisons between samples.</p>
<p>- Use appropriate methods for differential expression analysis: Recommended methods include limma (voom), DESeq2, and edgeR.</p>
<p>- Select suitable gene-level statistics: For FCS II, choose metrics like moderated t-statistic to rank genes meaningfully.</p>
<p>- Adjust for multiple testing: Ensure your analysis includes a correction for multiple hypothesis testing. Some methods require manual adjustments.</p>
<p>- Choose gene set databases based on biological context: Ensure that the database aligns with the research question and the experimental system.</p>
</div>
<div id="recommendation" class="section level3" number="7.5.3">
<h3>
<span class="header-section-number">7.5.3</span> <span style="color:red;">Recommendation</span><a class="anchor" aria-label="anchor" href="#recommendation"><i class="fas fa-link"></i></a>
</h3>
<ul>
<li>
<p>Awareness of Uncertainties:</p>
<p>- Recognise uncertainties in methods, parameter choices, and data preprocessing when conducting Gene Set Analysis (GSA).</p>
<p>- Understand that the method’s name alone does not capture the full analysis pipeline.</p>
</li>
<li><p>Clearly document all analysis choices, including methods, parameters, and preprocessing steps.</p></li>
<li><p>Select methods, parameters, and preprocessing steps before starting the analysis to minimise bias.</p></li>
<li>
<p>Set Technical Parameters:</p>
<p>- Fix technical parameters like the random seed and number of permutations before running the analysis to ensure reproducibility.</p>
<p>- Avoid adjusting these parameters to obtain favorable results.</p>
</li>
<li>
<p>Avoid Cherry-picking:</p>
<p>- Refrain from selectively reporting results based on favorable outcomes, as this can lead to over-optimistic and non-reproducible findings.</p>
<p>- Avoid excessive tweaking of the analysis strategy to fit the data post hoc.</p>
</li>
<li><p>Use different pipelines or parameter configurations as part of sensitivity analysis to check the consistency of results.</p></li>
<li><p>Share complete analysis workflows, including code and documentation, to allow others to replicate the findings accurately.</p></li>
</ul>
</div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="defining-the-genelist.html"><span class="header-section-number">6</span> Defining the genelist</a></div>
<div class="next"><a href="reporting-results.html"><span class="header-section-number">8</span> Reporting results</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#online-tools"><span class="header-section-number">7</span> Online Tools</a></li>
<li>
<a class="nav-link" href="#fea-in-gprofiler"><span class="header-section-number">7.1</span> FEA in gProfiler </a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#steps-to-perform-ora-in-gprofiler"><span class="header-section-number">7.1.1</span> Steps to perform ORA in g:Profiler:</a></li>
<li><a class="nav-link" href="#steps-to-perform-gsea-in-gprofiler"><span class="header-section-number">7.1.2</span> Steps to perform GSEA in g:Profiler:</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#fea-in-string"><span class="header-section-number">7.2</span> FEA in STRING </a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#steps-to-perform-ora-in-string"><span class="header-section-number">7.2.1</span> Steps to Perform ORA in STRING:</a></li>
<li><a class="nav-link" href="#browse-the-string-ora-results"><span class="header-section-number">7.2.2</span> Browse the STRING ORA Results</a></li>
<li><a class="nav-link" href="#steps-to-perform-gsea-in-string"><span class="header-section-number">7.2.3</span> Steps to Perform GSEA in STRING:</a></li>
<li><a class="nav-link" href="#browse-the-string-gsea-results"><span class="header-section-number">7.2.4</span> Browse the STRING GSEA Results</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#fea-in-genepattern"><span class="header-section-number">7.3</span> FEA in GenePattern </a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#steps-to-locate-gsea-module-in-genepattern"><span class="header-section-number">7.3.1</span> Steps to Locate GSEA Module in GenePattern:</a></li>
<li><a class="nav-link" href="#steps-to-perform-gsea"><span class="header-section-number">7.3.2</span> Steps to Perform GSEA:</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#fea-in-reactome"><span class="header-section-number">7.4</span> FEA in Reactome </a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#steps-to-perform-ora-in-reactome"><span class="header-section-number">7.4.1</span> Steps to perform ORA in Reactome:</a></li>
<li><a class="nav-link" href="#steps-to-perform-gsa-in-reactome"><span class="header-section-number">7.4.2</span> Steps to perform GSA in Reactome:</a></li>
<li><a class="nav-link" href="#browse-the-reactome-results"><span class="header-section-number">7.4.3</span> Browse the Reactome results</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#uncertainties-of-a-functional-enrichment-analsysis"><span class="header-section-number">7.5</span> Uncertainties of a functional enrichment analsysis</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#types-of-fea"><span class="header-section-number">7.5.1</span> Types of FEA</a></li>
<li><a class="nav-link" href="#considerations"><span class="header-section-number">7.5.2</span> Considerations</a></li>
<li><a class="nav-link" href="#recommendation"><span class="header-section-number">7.5.3</span> Recommendation</a></li>
</ul>
</li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://monashbioinformaticsplatform.github.io/Functional_Enrichment_Online_Tools//blob/master/06-web-tools.Rmd">View source <i class="fab fa-gitlab"></i></a></li>
          <li><a id="book-edit" href="https://monashbioinformaticsplatform.github.io/Functional_Enrichment_Online_Tools//edit/master/06-web-tools.Rmd">Edit this page <i class="fab fa-gitlab"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Functional Enrichment Workshop</strong>" was written by Australian Biocommons, Sydney Informatics Hub (USYD) and Monash Genomics and Bioinformatics Platform (MGBP). It was last built on <span>“October 16, 2025”</span> <br><img src="logos.png" width="300" alt="Logo"><br>.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
