<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 13 R environment set up | Functional Enrichment Workshop</title>
<meta name="author" content="Australian Biocommons, Sydney Informatics Hub (USYD) and Monash Genomics and Bioinformatics Platform (MGBP)">
<meta name="description" content="Making sense of gene and proteins lists with functional enrichment analysis   13.1 VM login You have previosuly been provided with an IP address, user name and password for a VM for this workshop....">
<meta name="generator" content="bookdown 0.40 with bs4_book()">
<meta property="og:title" content="Chapter 13 R environment set up | Functional Enrichment Workshop">
<meta property="og:type" content="book">
<meta property="og:description" content="Making sense of gene and proteins lists with functional enrichment analysis   13.1 VM login You have previosuly been provided with an IP address, user name and password for a VM for this workshop....">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 13 R environment set up | Functional Enrichment Workshop">
<meta name="twitter:description" content="Making sense of gene and proteins lists with functional enrichment analysis   13.1 VM login You have previosuly been provided with an IP address, user name and password for a VM for this workshop....">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.8.0/transition.js"></script><script src="libs/bs3compat-0.8.0/tabs.js"></script><script src="libs/bs3compat-0.8.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/kePrint-0.0.1/kePrint.js"></script><link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Functional Enrichment Workshop</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> Functional Enrichment Workshop</a></li>
<li class="book-part">Day 1</li>
<li><a class="" href="overview.html"><span class="header-section-number">3</span> Overview</a></li>
<li><a class="" href="recap.html"><span class="header-section-number">4</span> Recap</a></li>
<li><a class="" href="enrichment-statistics.html"><span class="header-section-number">5</span> Enrichment Statistics</a></li>
<li><a class="" href="example-analysis.html"><span class="header-section-number">6</span> Example Analysis</a></li>
<li><a class="" href="defining-the-genelist.html"><span class="header-section-number">7</span> Defining the genelist</a></li>
<li><a class="" href="online-tools.html"><span class="header-section-number">8</span> Online Tools</a></li>
<li><a class="" href="reporting-results.html"><span class="header-section-number">9</span> Reporting results</a></li>
<li><a class="" href="resources.html"><span class="header-section-number">10</span> Resources</a></li>
<li class="book-part">Day 2</li>
<li><a class="" href="introduction.html"><span class="header-section-number">12</span> Introduction</a></li>
<li><a class="active" href="r-environment-set-up.html"><span class="header-section-number">13</span> R environment set up</a></li>
<li><a class="" href="gprofiler2.html"><span class="header-section-number">14</span> gprofiler2</a></li>
<li><a class="" href="clusterprofiler-and-enrichplot.html"><span class="header-section-number">15</span> clusterProfiler and enrichplot</a></li>
<li><a class="" href="webgestaltr.html"><span class="header-section-number">16</span> WebGestaltR</a></li>
<li><a class="" href="novel-species-fea.html"><span class="header-section-number">17</span> Novel species FEA</a></li>
<li><a class="" href="workshop-wrap-up.html"><span class="header-section-number">18</span> Workshop wrap-up</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://monashbioinformaticsplatform.github.io/Functional_Enrichment_Online_Tools/">View book source <i class="fab fa-gitlab"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><span class="main-title"><em>Making sense of gene and proteins lists with functional enrichment analysis</em></span>
<div id="r-environment-set-up" class="section level1" number="13">
<h1>
<span class="header-section-number">13</span> R environment set up<a class="anchor" aria-label="anchor" href="#r-environment-set-up"><i class="fas fa-link"></i></a>
</h1>
<div id="vm-login" class="section level2" number="13.1">
<h2>
<span class="header-section-number">13.1</span> VM login<a class="anchor" aria-label="anchor" href="#vm-login"><i class="fas fa-link"></i></a>
</h2>
<p>You have previosuly been provided with an IP address, user name and password for a VM for this workshop. The VM runs RStudio and has all of the required R libraries for today‚Äôs workshop pre-installed.</p>
<p>
¬†
</p>
<p><!-- insert blank line --></p>
<p>‚û§ Open the VM by copying the IP address into a web browser.</p>
<p>When prompted, select <code>RStudio</code> from the options:</p>
<div class="inline-figure"><img src="images/vm-rstudio-open.png" style="border: none; box-shadow: none; background: none; width: 100%;"></div>
<p>
¬†
</p>
<p><!-- insert blank line --></p>
<p>Then enter your unique user ID (eg ‚Äúuser1‚Äù) and password previously provided (and select ‚ÄòStay signed in when browser closes‚Äô if you wish):</p>
<div class="inline-figure"><img src="images/vm-user-pw.png" style="border: none; box-shadow: none; background: none;"></div>
<p>
¬†
</p>
<p><!-- insert blank line --></p>
<p>When RStudio opens, you will see empty <code>Console</code>, <code>History</code> and <code>Plots</code> panes:</p>
<div class="inline-figure"><img src="images/rstudio-default-open.png" style="border: none; box-shadow: none; background: none; width: 100%;"></div>
<p>
¬†
</p>
<p><!-- insert blank line --></p>
<p>You can toggle to other panes, for example change <code>Console</code> to <code>Terminal</code> or change <code>Plots</code> to <code>Files</code>.</p>
<p>üìÅ Since your VM only runs RStudio, you will use the <code>Files</code> pane to navigate your workshop folder and view any output files we generate.</p>
<p>The <code>Console</code> contains the command prompt (<code>&gt;</code>) which is where R commands are entered.</p>
<p>
¬†
</p>
<p><!-- insert blank line --></p>
<p>‚û§ Test this out by copy pasting or typing the below command into your console then pressing the <code>enter</code> key. This will print the path of your current working directory:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>Your path will be something like <code>/home/user1</code>.</p>
<p>If you are unfamiliar with R, don‚Äôt worry because we will provide all the R code that you need to complete today‚Äôs workshop üòÉ</p>
<p>
¬†
</p>
<p><!-- insert blank line --></p>
<p>‚û§ Click on <code>Environment</code> to change the pane to list all active objects in your R session.</p>
<p>An object refers to things like imported datasets, R dataframes, variables, etc. For a new session, the environment is empty.</p>
<p>
¬†
</p>
<p><!-- insert blank line --></p>
<p>Let‚Äôs put something in there:</p>
<p>‚û§ Create an R variable called <code>name</code> and fill it with your name, then print it with the <code>cat</code> command:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">name</span> <span class="op">&lt;-</span> <span class="st">'Cali'</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="va">name</span><span class="op">)</span></span></code></pre></div>
<p>Note that the object <code>name</code> is now listed in the environment.</p>
<p>
¬†
</p>
<p><!-- insert blank line --></p>
</div>
<div id="r-notebooks" class="section level2" number="13.2">
<h2>
<span class="header-section-number">13.2</span> R notebooks<a class="anchor" aria-label="anchor" href="#r-notebooks"><i class="fas fa-link"></i></a>
</h2>
<p>Today we will not be entering R commands directly into the console like this. We will instead be using an R notebook.</p>
<p>Using notebooks in RStudio is a great way to save your code and comments, as well as have the code output display inside the notebook. Notebooks can be easily shared with others so they can run your analysis, and also rendered to HTML which is a neat way of saving a static copy of your work and presenting results to others.</p>
<p>
¬†
</p>
<p><!-- insert blank line --></p>
<p>‚û§ Open a new R notebook from the RStudio toolbar by selecting <code>File</code> ‚Üí <code>New file</code> ‚Üí <code>R Notebook</code>:</p>
<div class="inline-figure"><img src="images/rstudio-new-notebook.png" style="border: none; box-shadow: none; background: none; width: 100%;"></div>
<p>The new notebook opens in the <code>Editor</code> pane. It has a placeholder title and basic starting instructions.</p>
<p>
¬†
</p>
<p><!-- insert blank line --></p>
<div id="code-chunks" class="section level3" number="13.2.1">
<h3>
<span class="header-section-number">13.2.1</span> Code chunks<a class="anchor" aria-label="anchor" href="#code-chunks"><i class="fas fa-link"></i></a>
</h3>
<p>A code chunk is placed within triple backticks (```), and the language of the code is included on the first line, in this case <code>'r'</code>. Having code in chunks is a way of grouping and running related lines of code together.</p>
<p>Code chunks can be run by using the green arrow to the right of the chunk or by clicking inside the chunk and entering <code>ctrl + shift + enter</code>. There are additional run options under the <code>Run</code> menu from the top bar, for example <code>Run current chunk</code>, <code>Run next chunk</code>, or <code>Run all chunks below</code>.</p>
<p>Code chunks can also be given labels, by placing these inside the curly brackets, leaving a space after the <code>'r'</code>. Labels must be unique or rendering to HTML will fail with a message to correct the duplicated labels.</p>
<p>New code chunks can be added with the shortcut <code>ctrl + alt + i</code> or via the toolbar <code>Code</code> ‚Üí <code>Insert chunk</code>.</p>
<p>
¬†
</p>
<p><!-- insert blank line --></p>
<p>‚û§ Run the demo code chunk that was included in the new notebook to plot <code>cars</code></p>
<p>Note that the plot dispalys <em>inside</em> the notebook, rather than within the plot pane. If this code was executed directly from the <code>console</code> rather than the notebook, the plot would be in the <code>Plots</code> pane.</p>
<p>
¬†
</p>
<p><!-- insert blank line --></p>
</div>
<div id="rendered-html-notebooks" class="section level3" number="13.2.2">
<h3>
<span class="header-section-number">13.2.2</span> Rendered HTML notebooks<a class="anchor" aria-label="anchor" href="#rendered-html-notebooks"><i class="fas fa-link"></i></a>
</h3>
<p>Next we will look at what a HTML version of the notebook looks like. In order to preview the HTML, we first need to save the notebook.</p>
<p>
¬†
</p>
<p><!-- insert blank line --></p>
<p>‚û§ Change the title from the default of <code>"R Notebook"</code>to something of your choice, and make any other edits you want to, such as deleting some of the default content</p>
<p><strong>Note:</strong> don‚Äôt delete the <code>output: html_notebook</code> content from under the title! This will prevent the option to preview the html.</p>
<p>
¬†
</p>
<p><!-- insert blank line --></p>
<p>‚û§ Save the notebook (either <code>ctrl + s</code> or <code>File</code> ‚Üí <code>Save as</code>) then preview the HTML by selecting <code>Preview</code> from the editor pane toolbar</p>
<p><code>Preview</code> is for a quick check of how your notebook renders while working on it. For a fullly rendered, static document, use the <code>knit</code> function.</p>
<p>
¬†
</p>
<p><!-- insert blank line --></p>
<p>‚û§ Knit the notebook to HTML by selecting <code>Knit</code> ‚Üí <code>Knit to HTML</code> from the editor pane toolbar</p>
<p>Note that the HTML is saved to your current working directory, which we previously verified was <code>/home/userN</code>. Check that the file appears where you expect it to via the <code>Files</code> pane.</p>
<p>
¬†
</p>
<p><!-- insert blank line --></p>
</div>
<div id="a-fresh-workspace" class="section level3" number="13.2.3">
<h3>
<span class="header-section-number">13.2.3</span> A fresh workspace<a class="anchor" aria-label="anchor" href="#a-fresh-workspace"><i class="fas fa-link"></i></a>
</h3>
<p>Next we will open the R notebook for the first analysis activity. It‚Äôs ideal to start a new analysis with a clear environment, to avoid unintended object name clashes.</p>
<p>‚û§ Clear your environment by selecting <code>Session</code> ‚Üí <code>Quit session</code> ‚Üí <code>Dont save</code> ‚Üí<code>Start mew session</code></p>
<p><strong>Note:</strong> when asked <code>Save workspace image to ~/R.Data?</code> please select <strong><code>Don't Save</code></strong> during this workshop.</p>
<p>
¬†
</p>
<!-- insert blank line -->
<details><summary>
Workspace data: to save or not to save?
</summary><p>Not saving the workspace image can help avoid workspace clashes that can be hard to resolve or have unintended consequences. You don‚Äôt need to worry about losing data - after all, your input data is saved elsewhere, and your R code that produces all required outputs is safely saved within the notebook.</p>
<p>Saving the workspace image saves all objects from the session such as your variables and dataframes. This can save time if you need to close an analysis part way through and continue later. However, this can have drawbacks such as library and function name clashes, unexpected objects present in the environment, large objects and relic objects cluttering the workspace, old objects conflicting with new ones, etc. Since we will be performing discrete analysis tasks today, and not continuing on a growing body of work, selecting <code>Dont save</code> will be the most appropriate.</p>
</details><p>
¬†
</p>
<p><!-- insert blank line --></p>
<p>üëÄ You may notice there is also a <code>Clear Workspace</code> option under <code>Session</code>. This will remove all R objects from your environment, but it won‚Äôt remove loaded libraries. You can of course unload these with R code, but refreshing the session is easier üòä Some libraries share function names. If you are being very correct, you can prefix the R package before the function, to ensure the exact function you want is being called to avoid any potential function clashes. I admit to being guilty of not doing that enough! ü§≠</p>
<p>
¬†
</p>
<p><!-- insert blank line --></p>
</div>
</div>
<div id="download-input-data" class="section level2" number="13.3">
<h2>
<span class="header-section-number">13.3</span> Download input data<a class="anchor" aria-label="anchor" href="#download-input-data"><i class="fas fa-link"></i></a>
</h2>
<p>‚û§ Copy the below code into your RStudio <code>console</code> and press the <code>enter</code> key</p>
<p>This will download <code>workshop.zip</code> and unzip it. This contains four R code notebooks (.Rmd) and six input data files.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html">download.file</a></span><span class="op">(</span><span class="st">"https://raw.githubusercontent.com/MonashBioinformaticsPlatform/Functional_Enrichment_BioCommons_2024/refs/heads/main/workshop.zip"</span>, <span class="st">"workshop.zip"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/unzip.html">unzip</a></span><span class="op">(</span><span class="st">"workshop.zip"</span><span class="op">)</span></span></code></pre></div>
<p>
¬†
</p>
<p><!-- insert blank line --></p>
<p>Your <code>workshop</code> folder should now look like this, with a total of 10 files:</p>
<div class="inline-figure"><img src="images/workshop-folder.png" style="border: none; box-shadow: none; background: none;"></div>
<p>
¬†
</p>
<p><!-- insert blank line --></p>
<p>‚úã <strong>Please ask for assistance if you have had any issues downloading the input data.</strong></p>
<p>
¬†
</p>
<p><!-- insert blank line --></p>
<div id="working-directory" class="section level3" number="13.3.1">
<h3>
<span class="header-section-number">13.3.1</span> Working directory<a class="anchor" aria-label="anchor" href="#working-directory"><i class="fas fa-link"></i></a>
</h3>
<p>Now that we have a clear workspace, we will prepare for the first analysis activity by opening the code notebook and checking our working directory.</p>
<p>‚û§ Load the notebook <code>gprofiler2.Rmd</code> notebook by clicking on it in the <code>Files</code> pane</p>
<p>You could also open the file by selecting <code>File</code> ‚Üí <code>Open file</code>, or use the keyboard shortcut <code>ctrl + o</code>.</p>
<p>
¬†
</p>
<p><!-- insert blank line --></p>
<p>The default working directory for an R notebook is the directory in which it is saved.</p>
<p>‚û§ Confirm that your working directory is the <code>workshop</code> folder by running the code chunk labelled <code>check notebook workdir</code> (use the green arrow to the right of the chunk)</p>
<p>Since our code notebooks and input data files are saved within an overly simple directory structure, we won‚Äôt need to modify the working directory. However in practice, you would typically have different working directories or nested directories, so changing directories and/or providing directory paths when loading or saving files would be required.</p>
<p>The code chunk labelled <code>working directory</code> contains only ‚Äúhashed out‚Äù code and is merely a placeholder for if a directory change is required.</p>
<p>
¬†
</p>
<p><!-- insert blank line --></p>
<p>Note that the default directory for RStusio (set in the ‚ÄòGlobal options‚Äô) and the default directory for a code notebook differ! You can change the working directory of the console to match the notebook by issuing the below command into your console directly:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/getwd.html">setwd</a></span><span class="op">(</span><span class="st">"workshop"</span><span class="op">)</span></span></code></pre></div>
<p>
¬†
</p>
<p><!-- insert blank line --></p>
<p>Scroll down to the code chunk labelled <code>load input data</code>. Note that the filepath is simply <code>Pezzini_DE.txt</code>. We don‚Äôt need to specify the full path to our working directory, only the relative path. Since the notebook working directory and input data are both within the same directory, we can simply load the input file by providing the fle name.</p>
<p>
¬†
</p>
<p><!-- insert blank line --></p>
</div>
<div id="r-packages" class="section level3" number="13.3.2">
<h3>
<span class="header-section-number">13.3.2</span> R packages<a class="anchor" aria-label="anchor" href="#r-packages"><i class="fas fa-link"></i></a>
</h3>
<p>Immediately above the <code>check notebook workdir</code> code chunk is a code chunk labelled <code>Load R packages</code>. This contains all of the R packages required to run the analysis contained within the workbook. Loading all required packages within the notebook, rather than directly via the console, ensures that anyone running your notebook does not encounter errors if they forget to load a required package.</p>
<p>Note that the packages that are loaded to the session with the R <code>library</code> command must first be installed; this has already been been done for you on these VMs. Attempts to load a package that is not installed will produce an error, and installation can then be peformed (not difficult in R) before resuming.</p>
<p>Note that the code chunk label also contains the text <code>include=FALSE</code>. This prevents the loading of libraries (which can at times have verbose output) from cluttering up your rendered notebook when it is previewed or knit.</p>
<p>
¬†
</p>
<p><!-- insert blank line --></p>
<p>‚û§ Run the <code>Load R packages</code> code chunk.</p>
<p>
¬†
</p>
<p><!-- insert blank line --></p>
<p>‚úã <strong>Please ask for assistance if you have had any issues loading the R libraries</strong></p>
<p>Don‚Äôt be alarmed that the output is <span style="color: red;">red</span>! üôÇ</p>

</div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="introduction.html"><span class="header-section-number">12</span> Introduction</a></div>
<div class="next"><a href="gprofiler2.html"><span class="header-section-number">14</span> gprofiler2</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#r-environment-set-up"><span class="header-section-number">13</span> R environment set up</a></li>
<li><a class="nav-link" href="#vm-login"><span class="header-section-number">13.1</span> VM login</a></li>
<li>
<a class="nav-link" href="#r-notebooks"><span class="header-section-number">13.2</span> R notebooks</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#code-chunks"><span class="header-section-number">13.2.1</span> Code chunks</a></li>
<li><a class="nav-link" href="#rendered-html-notebooks"><span class="header-section-number">13.2.2</span> Rendered HTML notebooks</a></li>
<li><a class="nav-link" href="#a-fresh-workspace"><span class="header-section-number">13.2.3</span> A fresh workspace</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#download-input-data"><span class="header-section-number">13.3</span> Download input data</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#working-directory"><span class="header-section-number">13.3.1</span> Working directory</a></li>
<li><a class="nav-link" href="#r-packages"><span class="header-section-number">13.3.2</span> R packages</a></li>
</ul>
</li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://monashbioinformaticsplatform.github.io/Functional_Enrichment_Online_Tools//blob/master/10-r-environment-setup.Rmd">View source <i class="fab fa-gitlab"></i></a></li>
          <li><a id="book-edit" href="https://monashbioinformaticsplatform.github.io/Functional_Enrichment_Online_Tools//edit/master/10-r-environment-setup.Rmd">Edit this page <i class="fab fa-gitlab"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Functional Enrichment Workshop</strong>" was written by Australian Biocommons, Sydney Informatics Hub (USYD) and Monash Genomics and Bioinformatics Platform (MGBP). It was last built on <span>‚ÄúOctober 14, 2025‚Äù</span> <br><img src="logos.png" width="300" alt="Logo"><br>.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
