<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 3 Recap | Functional Enrichment Workshop</title>
<meta name="author" content="Australian Biocommons, Sydney Informatics Hub (USYD) and Monash Genomics and Bioinformatics Platform (MGBP)">
<meta name="description" content="Making sense of gene and proteins lists with functional enrichment analysis   3.1 Functional enrichment analysis Functional enrichment analysis (FEA) refers to a set of computational approaches...">
<meta name="generator" content="bookdown 0.40 with bs4_book()">
<meta property="og:title" content="Chapter 3 Recap | Functional Enrichment Workshop">
<meta property="og:type" content="book">
<meta property="og:description" content="Making sense of gene and proteins lists with functional enrichment analysis   3.1 Functional enrichment analysis Functional enrichment analysis (FEA) refers to a set of computational approaches...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 3 Recap | Functional Enrichment Workshop">
<meta name="twitter:description" content="Making sense of gene and proteins lists with functional enrichment analysis   3.1 Functional enrichment analysis Functional enrichment analysis (FEA) refers to a set of computational approaches...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.8.0/transition.js"></script><script src="libs/bs3compat-0.8.0/tabs.js"></script><script src="libs/bs3compat-0.8.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/kePrint-0.0.1/kePrint.js"></script><link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Functional Enrichment Workshop</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> Functional Enrichment Workshop</a></li>
<li><a class="" href="overview.html"><span class="header-section-number">2</span> Overview</a></li>
<li><a class="active" href="recap.html"><span class="header-section-number">3</span> Recap</a></li>
<li><a class="" href="enrichment-statistics.html"><span class="header-section-number">4</span> Enrichment Statistics</a></li>
<li><a class="" href="example-analysis.html"><span class="header-section-number">5</span> Example Analysis</a></li>
<li><a class="" href="defining-the-genelist.html"><span class="header-section-number">6</span> Defining the genelist</a></li>
<li><a class="" href="online-tools.html"><span class="header-section-number">7</span> Online Tools</a></li>
<li><a class="" href="reporting-results.html"><span class="header-section-number">8</span> Reporting results</a></li>
<li><a class="" href="resources.html"><span class="header-section-number">9</span> Resources</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://monashbioinformaticsplatform.github.io/Functional_Enrichment_Online_Tools/">View book source <i class="fab fa-gitlab"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><span class="main-title"><em>Making sense of gene and proteins lists with functional enrichment analysis</em></span>
<div id="recap" class="section level1" number="3">
<h1>
<span class="header-section-number">3</span> Recap<a class="anchor" aria-label="anchor" href="#recap"><i class="fas fa-link"></i></a>
</h1>
<div id="functional-enrichment-analysis" class="section level2" number="3.1">
<h2>
<span class="header-section-number">3.1</span> Functional enrichment analysis<a class="anchor" aria-label="anchor" href="#functional-enrichment-analysis"><i class="fas fa-link"></i></a>
</h2>
<p>Functional enrichment analysis (FEA) refers to a set of computational approaches designed to derive biological meaning from lists of biomolecules, such as genes, proteins, or metabolites.</p>
<p>By focusing on the biological significance of biomolecular changes, functional enrichment analysis enables researchers to make sense of complex high-throughput data from large-scale studies, revealing the key cellular processes and signaling pathways involved in health and disease states.</p>
<div id="why-is-it-important" class="section level3" number="3.1.1">
<h3>
<span class="header-section-number">3.1.1</span> Why Is It Important?<a class="anchor" aria-label="anchor" href="#why-is-it-important"><i class="fas fa-link"></i></a>
</h3>
<p>Large-scale omics studies often yield vast datasets with hundreds or thousands of significantly regulated biomolecules. Manually investigating each feature, such as individual genes or proteins, can be overwhelming and inefficient.</p>
<p>Functional enrichment analysis provides a solution by organising these biomolecules into meaningful categories, allowing for the identification of overarching biological patterns and mechanisms. This helps reduce data complexity and uncovers higher-level biological insights, such as discovering critical pathways involved in disease progression or identifying potential therapeutic targets. Thus, enrichment analysis is a crucial step in the interpretation of high-dimensional omics data, transforming lists of molecular entities into actionable biological knowledge.</p>
</div>
<div id="when-to-use-functional-enrichment-analysis" class="section level3" number="3.1.2">
<h3>
<span class="header-section-number">3.1.2</span> When to Use Functional Enrichment Analysis?<a class="anchor" aria-label="anchor" href="#when-to-use-functional-enrichment-analysis"><i class="fas fa-link"></i></a>
</h3>
<p>Functional enrichment analysis is typically applied after conducting a differential expression analysis or other comparative analyses in omics studies. This step is essential when attempting to derive biological insights from large lists of biomolecules that exhibit significant changes in expression, modification, or abundance between experimental groups. Below are common scenarios where functional enrichment analysis is particularly valuable:</p>
<ul>
<li>Transcriptomics (eg high-fat diet vs. low-fat diet)</li>
<li>Proteomics (eg tumor tissue vs. healthy tissue)</li>
<li>Metabolomics (eg diabetic vs. non-diabetic patients)</li>
<li>Lipidomics (eg disease vs. healthy state)</li>
<li>Epigenomics (eg smokers vs. non-smokers)</li>
<li>…</li>
</ul>
</div>
<div id="what-are-the-input-data" class="section level3" number="3.1.3">
<h3>
<span class="header-section-number">3.1.3</span> What Are the Input Data?<a class="anchor" aria-label="anchor" href="#what-are-the-input-data"><i class="fas fa-link"></i></a>
</h3>
<p>Functional enrichment analysis relies on carefully prepared input data derived from an -omics study. The data inputs typically consist of various components depending on the type of the enrichment analysis. These are list of features, ranked list, background set, gene sets and pathway topology.</p>
</div>
<div id="synonyms" class="section level3" number="3.1.4">
<h3>
<span class="header-section-number">3.1.4</span> Synonyms<a class="anchor" aria-label="anchor" href="#synonyms"><i class="fas fa-link"></i></a>
</h3>
<p>It’s important to note that the term “functional enrichment analysis” is often used in different ways across the field. The diversity in terminology can sometimes cause confusion, as the same concept is referred to by various synonymous terms. These include:</p>
<ul>
<li>Enrichment analysis</li>
<li>Pathway analysis</li>
<li>Pathway enrichment analysis</li>
<li>Functional annotation analysis</li>
<li>Annotation enrichment analysis</li>
<li>Functional pathway analysis</li>
<li>Functional enrichment analysis</li>
<li>…</li>
</ul>
</div>
</div>
<div id="concepts" class="section level2" number="3.2">
<h2>
<span class="header-section-number">3.2</span> Concepts<a class="anchor" aria-label="anchor" href="#concepts"><i class="fas fa-link"></i></a>
</h2>
<p>Here we will explore key concepts that are essential to perform and interpret enrichment analysis, including gene lists, background sets, p-values, false discovery rates, and the role of annotation databases. These concepts form the foundation for making sense of the biological significance of experimental findings.</p>
<ul>
<li><p><span style="color: orange;">Gene List:</span>
A gene list is the collection of genes (or proteins) that are of particular interest in a biological experiment. This list typically arises from high-throughput experiments such as transcriptomics, proteomics, or genomics, where genes are differentially expressed, mutated, or otherwise identified as significant. In functional enrichment analysis, the gene list is used to assess whether certain biological pathways, gene ontologies, or functions are statistically overrepresented compared to a reference or background set.</p></li>
<li><p><span style="color: orange;">Background Set:</span>
The background set, also referred to as the “reference set,” is the complete set of genes or proteins against which the gene list is compared. This background typically includes all genes that were analysed in the experiment (e.g., all genes in a microarray or RNA-seq dataset). The choice of background is crucial because it influences the statistical significance of the enrichment. For instance, using a background that includes only expressed genes will result in a different outcome compared to using all known genes in the genome.</p></li>
<li><p><span style="color: orange;">P-value:</span>
The p-value is a measure of the probability that the observed result occurred by chance. At the feature level, it indicates whether a particular gene or protein shows significant differences (e.g., in expression or mutation) when compared to a control or baseline. For example, a p-value of 0.01 for a gene means there’s only a 1% chance that the observed change in that gene is due to random variation. At the enrichment level, the p-value evaluates whether the overlap between the identified genes and a particular biological term (such as a pathway) occurred by chance. A pathway with a p-value of 0.001, for instance, suggests that there’s only a 0.1% probability that the pathway’s association with the gene list occurred randomly.</p></li>
<li><p><span style="color: orange;">False Discovery Rate (FDR):</span>
The FDR corrects for multiple comparisons, as many tests are conducted both at the feature and enrichment levels. When analysing thousands of genes and numerous pathways, the likelihood of false positives increases, so FDR adjusts for this by controlling the proportion of false positives among the significant results. For example, an FDR threshold of 0.05 means that no more than 5% of the features (e.g., genes) or enriched terms (e.g., pathways) identified as significant are expected to be false positives.</p></li>
<li><p><span style="color: orange;">Regulation:</span>
In the context of enrichment analysis, regulation refers to the upregulation or downregulation of genes. Many enrichment tools allow users to analyse gene lists with regulation status included. This means pathways or biological functions may be enriched with genes that are specifically upregulated (increased activity) or downregulated (decreased activity). This additional layer of information helps in understanding whether certain pathways or processes are being activated or suppressed in the condition of interest.</p></li>
<li><p><span style="color: orange;">ID Mapping:</span>
ID mapping refers to the process of converting different types of gene or protein identifiers into a unified format. This is necessary because different databases and platforms may use different types of identifiers (e.g., gene symbols, Entrez IDs, Ensembl IDs, Uniprot IDs). Accurate ID mapping ensures that the gene list aligns with the annotation database being used in the analysis. Tools and databases often provide built-in options for ID conversion to facilitate this step.</p></li>
<li><p><span style="color: orange;">Annotation Databases:</span>
An annotation database is a curated collection of biological data that links genes or proteins to functional information such as pathways, molecular functions, cellular components, and biological processes. Examples include Gene Ontology (GO), KEGG, Reactome, and MSigDB. These databases provide the functional terms or pathways that are tested for enrichment. The choice of annotation database can significantly influence the results, as different databases may focus on different types of biological information or contain slightly different gene-function relationships.</p></li>
</ul>
</div>
<div id="types-of-enrichment-analysis" class="section level2" number="3.3">
<h2>
<span class="header-section-number">3.3</span> Types of Enrichment Analysis<a class="anchor" aria-label="anchor" href="#types-of-enrichment-analysis"><i class="fas fa-link"></i></a>
</h2>
<p>Khatri et al. (2012) nicely explained different types of enrichment analysis, as shown below.</p>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-5"></span>
<img src="images/fea_types.jpg" alt="Types of  of functional enrichment analysis" width="100%"><p class="caption">
Figure 3.1: Types of of functional enrichment analysis
</p>
</div>
<p><em>Source: Figure adapted from Khatri P, Sirota M, Butte AJ. Ten years of pathway analysis: current approaches and outstanding challenges. PLoS Comput Biol. 2012;8(2):e1002375.</em></p>
<div id="over-representation-analysis-ora" class="section level3" number="3.3.1">
<h3>
<span class="header-section-number">3.3.1</span> Over Representation Analysis (ORA)<a class="anchor" aria-label="anchor" href="#over-representation-analysis-ora"><i class="fas fa-link"></i></a>
</h3>
<p>Over Representation Analysis (ORA) is one of the simplest and most widely used methods for functional enrichment analysis. ORA aims to determine whether specific biological categories (e.g., pathways, Gene Ontology terms) are statistically overrepresented in a given list of features (like genes or proteins) compared to a background or reference set. This method focuses on counting the number of features from the list that are associated with a specific category and comparing this count to what would be expected by chance.</p>
<div id="input-data" class="section level4" number="3.3.1.1">
<h4>
<span class="header-section-number">3.3.1.1</span> Input Data<a class="anchor" aria-label="anchor" href="#input-data"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li><p><span style="color: orange;">List of Features:</span> This refers to the subset of biomolecules identified as significantly regulated or altered in the study. Features might include genes, proteins, lipids, or other biomolecules, depending on the type of -omics data.</p></li>
<li><p><span style="color: orange;">Background Set:</span> The background set, or universe, consists of all the features that were measured in the study or a defined subset of the total genome, proteome, or metabolome being studied. The background is critical for enrichment analysis because it provides the context against which the significance of feature enrichment is assessed.</p></li>
</ul>
</div>
<div id="workflow" class="section level4" number="3.3.1.2">
<h4>
<span class="header-section-number">3.3.1.2</span> Workflow<a class="anchor" aria-label="anchor" href="#workflow"><i class="fas fa-link"></i></a>
</h4>
<p><span style="color: orange;"><em>How it works</em></span>: ORA uses a predefined feature list (e.g., from differentially expressed genes), calculates the number of features in the list that belong to a certain category (e.g., a pathway), and tests whether this number is significantly higher than expected using statistical tests like the hypergeometric test or Fisher’s exact test.</p>
<p><span style="color: orange;"><em>Strengths</em></span>: Simple and easy to implement. Works well with a predefined list of significant features.</p>
<p><span style="color: orange;"><em>Limitations</em></span>: ORA does not take into account the full range of feature expression values and can miss subtle changes across a broader set of features. It relies heavily on selecting a predefined cut-off to create the feature list, which can be subjective.</p>
</div>
</div>
<div id="gene-set-enrichment-analysis-gsea" class="section level3" number="3.3.2">
<h3>
<span class="header-section-number">3.3.2</span> Gene Set Enrichment Analysis (GSEA)<a class="anchor" aria-label="anchor" href="#gene-set-enrichment-analysis-gsea"><i class="fas fa-link"></i></a>
</h3>
<p>Gene Set Enrichment Analysis (GSEA) also known as Functional Class Scoring (FCS) is a more sophisticated method that avoids the need to define a strict cut-off for selecting a list of significant features. Instead of using a discrete list of differentially expressed features, GSEA analyses ranked feature expression data. It evaluates whether predefined gene sets (such as pathways or functional categories) are enriched at the top or bottom of the ranked list, capturing subtle but coordinated changes in gene expression.</p>
<div id="input-data-1" class="section level4" number="3.3.2.1">
<h4>
<span class="header-section-number">3.3.2.1</span> Input Data<a class="anchor" aria-label="anchor" href="#input-data-1"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li><p><span style="color: orange;">Ranked List:</span> In some enrichment methods, such as Gene Set Enrichment Analysis (GSEA), a ranked list is used instead of a simple feature list. The ranking is typically based on a continuous metric such as the magnitude of gene expression changes or some sort of statistical test output. This ranked list helps prioritise features that exhibit the strongest biological relevance and facilitates more nuanced enrichment analyses that consider the direction and strength of biomolecular changes.</p></li>
<li><p><span style="color: orange;">Gene Sets:</span> This refers to predefined groups of genes that share a common biological property, such as involvement in a specific biological pathway, functional category, or regulatory process. The most commonly used format for gene sets in MSigDB is the GMT format. This format is simple, human-readable, and widely supported by various GSEA tools.</p></li>
</ul>
</div>
<div id="workflow-1" class="section level4" number="3.3.2.2">
<h4>
<span class="header-section-number">3.3.2.2</span> Workflow<a class="anchor" aria-label="anchor" href="#workflow-1"><i class="fas fa-link"></i></a>
</h4>
<p><span style="color: orange;"><em>How it works</em></span>: GSEA first ranks all genes in the dataset according to their differential expression levels (e.g., from a control to a condition). Then, for each predefined gene set, it calculates an enrichment score (ES) that reflects the concentration of the gene set members at the extremes of the ranked list. Statistical significance is assessed through permutation testing, and the False Discovery Rate (FDR) is used to correct for multiple comparisons.</p>
<p><span style="color: orange;"><em>Strengths</em></span>: GSEA avoids arbitrary thresholds for feature selection and can detect coordinated changes across sets of genes, even if individual genes within the set do not show significant differential expression.</p>
<p><span style="color: orange;"><em>Limitations</em></span>: GSEA may miss smaller pathways or functional categories if their features are not highly ranked or uniformly expressed. It is also more computationally intensive than ORA.</p>
</div>
</div>
<div id="pathway-topology-pt-based-enrichment" class="section level3" number="3.3.3">
<h3>
<span class="header-section-number">3.3.3</span> Pathway Topology (PT)-Based Enrichment<a class="anchor" aria-label="anchor" href="#pathway-topology-pt-based-enrichment"><i class="fas fa-link"></i></a>
</h3>
<p>Pathway Topology (PT)-based enrichment analysis extends beyond merely counting features and instead incorporates the topological structure of biological pathways. This method evaluates not only which features are part of a pathway but also their position and interactions within the pathway. By considering the connectivity and interaction strength between features, PT-based approaches provide a more biologically meaningful interpretation of pathway activation or suppression.</p>
<div id="input-data-2" class="section level4" number="3.3.3.1">
<h4>
<span class="header-section-number">3.3.3.1</span> Input Data<a class="anchor" aria-label="anchor" href="#input-data-2"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li>
<span style="color: orange;">List of Features or Ranked List:</span> Already explained.</li>
<li>
<span style="color: orange;">Pathway Topology:</span> This refers to the structure of a biological pathway, which includes detailed information about the interactions and relationships between gene products (such as proteins or RNAs) within a pathway.</li>
</ul>
</div>
<div id="workflow-2" class="section level4" number="3.3.3.2">
<h4>
<span class="header-section-number">3.3.3.2</span> Workflow<a class="anchor" aria-label="anchor" href="#workflow-2"><i class="fas fa-link"></i></a>
</h4>
<p><span style="color: orange;"><em>How it works</em></span>: PT-based methods take into account the direction and magnitude of feature expression changes, as well as the structure of pathways (e.g., signaling cascades, metabolic pathways). They consider how biomolecule products interact with one another and the specific roles of each gene within the pathway. Topological factors like the number of connections a gene has or its centrality in the pathway are considered when assessing the enrichment.</p>
<p><span style="color: orange;"><em>Strengths</em></span>: Provides more biologically relevant insights by considering gene-gene interactions and the position of each gene within a pathway. It is particularly useful for complex pathways where the roles of genes differ based on their interactions with others.</p>
<p><span style="color: orange;"><em>Limitations</em></span>: Requires more detailed pathway annotations and higher computational complexity. Pathway databases may not have complete or accurate topological information for all pathways, limiting the analysis for certain datasets.</p>
<p>PT-based enrichment will be covered in this workshop:
Given the focus of this workshop on more widely used and accessible enrichment methods, PT-based analysis will not be covered for its limited practical applications (primarily due to the insufficient availability of comprehensive and well-annotated pathway topology databases). Instead, we will focus on methods like ORA and GSEA, which are better supported by existing databases and easier to apply in typical omics studies. However, participants are encouraged to explore PT-based enrichment in the future as database resources improve.</p>
</div>
</div>
</div>
<div id="annotation-databses" class="section level2" number="3.4">
<h2>
<span class="header-section-number">3.4</span> Annotation Databses<a class="anchor" aria-label="anchor" href="#annotation-databses"><i class="fas fa-link"></i></a>
</h2>
<p>Functional annotation databases are curated collections of biological data that systematically categorise and describe the functions, roles, interactions, and pathways of genes, proteins, or other biological molecules, enabling researchers to link experimental data to biological knowledge.</p>
<div id="go-gene-ontology" class="section level3" number="3.4.1">
<h3>
<span class="header-section-number">3.4.1</span> <a href="https://geneontology.org/">GO: Gene Ontology</a><a class="anchor" aria-label="anchor" href="#go-gene-ontology"><i class="fas fa-link"></i></a>
</h3>
<p>“The goal of the Gene Ontology Consortium is to produce a dynamic, controlled vocabulary that can be applied to all eukaryotes even as knowledge of gene and protein roles in cells is accumulating and changing.” (Ashburner et al. 2000)</p>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-6"></span>
<img src="images/go_structure.png" alt="The structure of gene ontology." width="100%"><p class="caption">
Figure 3.2: The structure of gene ontology.
</p>
</div>
<p><em>Imagen source <a href="https://link.springer.com/book/10.1007/978-1-4939-3743-1">The Gene Ontology Handbook</a></em></p>
<p>Gene Ontology (GO) is a structured framework used to describe the roles of genes and their products across all living organisms. It provides a controlled vocabulary that allows for consistent descriptions of gene functions, biological processes, and cellular locations, facilitating computational analysis and integration of biological data across different species. GO’s structure comprises three main aspects:</p>
<p><span style="color: orange;">1. Molecular Function:</span>
In the Gene Ontology (GO), molecular function refers to the specific biochemical activity that a gene product (such as a protein or RNA) performs. This activity typically involves direct physical interactions with other molecular entities, such as catalysis or binding. These functions are described based on their biochemical roles (e.g., enzyme activity) and their contribution as components within larger biological systems. For instance, protein kinase activity involves the phosphorylation of proteins, which is a specific molecular function. In GO, molecular function is concerned with the direct action of gene products, whether in terms of biochemical interactions or roles in larger biological systems.</p>
<p><span style="color: orange;">2. Biological Process:</span>
Biological processes represent the larger objectives that gene products contribute to in an organism, often described by the outcome or result of a series of molecular events. These processes are broader, coordinated sequences of molecular activities that achieve a biological objective, such as cell division or DNA replication. A biological process in GO can encompass anything from simple enzymatic actions to complex, regulated systems like embryonic development or immune response. GO annotations aim to associate gene products not only with the processes they directly contribute to but also with processes they regulate or enable.</p>
<p><span style="color: orange;">3. Cellular Component:</span>
This aspect of GO refers to the specific location within a cell where a gene product operates. Cellular components are described relative to structures within the cell, such as the mitochondrion or plasma membrane, and reflect where molecular functions occur as part of broader biological processes. These locations are vital to understanding where molecular activities take place, as cellular compartmentalisation often influences the function and regulation of gene products. Unlike molecular function and biological process, cellular components refer more to cellular anatomy, specifying where gene products perform their roles during biological activities.</p>
<p>In practice, GO terms and annotations allow researchers to describe gene functions in a standardised way, helping in tasks such as gene function prediction, functional profiling, and comparing genes across species. GO’s hierarchical organisation of terms provides a rich framework to model the complexity of biological systems and facilitates the computational study of gene functions.</p>
</div>
<div id="kegg-kyoto-encyclopedia-of-genes-and-genomes" class="section level3" number="3.4.2">
<h3>
<span class="header-section-number">3.4.2</span> <a href="https://www.genome.jp/kegg/">KEGG: Kyoto Encyclopedia of Genes and Genomes</a><a class="anchor" aria-label="anchor" href="#kegg-kyoto-encyclopedia-of-genes-and-genomes"><i class="fas fa-link"></i></a>
</h3>
<p>Kyoto Encyclopedia of Genes and Genomes (KEGG) is a curated database that integrates genomic, chemical, and systemic information to represent biological systems and their interactions. It allows users to map molecular data (such as genes, proteins, and small molecules) to biological pathways, enabling a better understanding of how different components interact within an organism.</p>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-7"></span>
<img src="images/NOTCH_signaling_pathway_kegg.png" alt="NOTCH Signaling Pathway by KEGG" width="100%"><p class="caption">
Figure 3.3: NOTCH Signaling Pathway by KEGG
</p>
</div>
</div>
<div id="reactome" class="section level3" number="3.4.3">
<h3>
<span class="header-section-number">3.4.3</span> <a href="https://reactome.org/">Reactome</a><a class="anchor" aria-label="anchor" href="#reactome"><i class="fas fa-link"></i></a>
</h3>
<p>Reactome pathway knowledgebase is an open-access, manually curated database that captures molecular details of biological processes such as signal transduction, DNA replication, metabolism, and more, using a consistent data model across different domains of biology. This makes it particularly well-suited for functional enrichment analysis, where understanding the relationships between gene expression data and biological pathways is crucial.</p>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-8"></span>
<img src="images/NOTCH_signaling_pathway_reactome.png" alt="NOTCH Signaling Pathway by Reactome" width="100%"><p class="caption">
Figure 3.4: NOTCH Signaling Pathway by Reactome
</p>
</div>
</div>
<div id="msigdb" class="section level3" number="3.4.4">
<h3>
<span class="header-section-number">3.4.4</span> <a href="https://www.gsea-msigdb.org/">MSigDB</a><a class="anchor" aria-label="anchor" href="#msigdb"><i class="fas fa-link"></i></a>
</h3>
<p>Molecular Signatures Database (MSigDB) is a comprehensive resource for gene set enrichment analysis. It offers a comprehensive collection of gene sets that represent biological processes, molecular pathways, and other biologically relevant information. MSigDB is integrated with the Gene Set Enrichment Analysis (GSEA) tool, which is commonly used to determine if predefined sets of genes show statistically significant differences between two biological states (e.g., diseased vs. healthy samples).</p>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-9"></span>
<img src="images/GSEA-homegraphic.gif" alt="GSEA Workflow" width="100%"><p class="caption">
Figure 3.5: GSEA Workflow
</p>
</div>
</div>
</div>
<div id="common-tools-for-doing-fea" class="section level2" number="3.5">
<h2>
<span class="header-section-number">3.5</span> Common Tools for Doing FEA<a class="anchor" aria-label="anchor" href="#common-tools-for-doing-fea"><i class="fas fa-link"></i></a>
</h2>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="overview.html"><span class="header-section-number">2</span> Overview</a></div>
<div class="next"><a href="enrichment-statistics.html"><span class="header-section-number">4</span> Enrichment Statistics</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#recap"><span class="header-section-number">3</span> Recap</a></li>
<li>
<a class="nav-link" href="#functional-enrichment-analysis"><span class="header-section-number">3.1</span> Functional enrichment analysis</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#why-is-it-important"><span class="header-section-number">3.1.1</span> Why Is It Important?</a></li>
<li><a class="nav-link" href="#when-to-use-functional-enrichment-analysis"><span class="header-section-number">3.1.2</span> When to Use Functional Enrichment Analysis?</a></li>
<li><a class="nav-link" href="#what-are-the-input-data"><span class="header-section-number">3.1.3</span> What Are the Input Data?</a></li>
<li><a class="nav-link" href="#synonyms"><span class="header-section-number">3.1.4</span> Synonyms</a></li>
</ul>
</li>
<li><a class="nav-link" href="#concepts"><span class="header-section-number">3.2</span> Concepts</a></li>
<li>
<a class="nav-link" href="#types-of-enrichment-analysis"><span class="header-section-number">3.3</span> Types of Enrichment Analysis</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#over-representation-analysis-ora"><span class="header-section-number">3.3.1</span> Over Representation Analysis (ORA)</a></li>
<li><a class="nav-link" href="#gene-set-enrichment-analysis-gsea"><span class="header-section-number">3.3.2</span> Gene Set Enrichment Analysis (GSEA)</a></li>
<li><a class="nav-link" href="#pathway-topology-pt-based-enrichment"><span class="header-section-number">3.3.3</span> Pathway Topology (PT)-Based Enrichment</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#annotation-databses"><span class="header-section-number">3.4</span> Annotation Databses</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#go-gene-ontology"><span class="header-section-number">3.4.1</span> GO: Gene Ontology</a></li>
<li><a class="nav-link" href="#kegg-kyoto-encyclopedia-of-genes-and-genomes"><span class="header-section-number">3.4.2</span> KEGG: Kyoto Encyclopedia of Genes and Genomes</a></li>
<li><a class="nav-link" href="#reactome"><span class="header-section-number">3.4.3</span> Reactome</a></li>
<li><a class="nav-link" href="#msigdb"><span class="header-section-number">3.4.4</span> MSigDB</a></li>
</ul>
</li>
<li><a class="nav-link" href="#common-tools-for-doing-fea"><span class="header-section-number">3.5</span> Common Tools for Doing FEA</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://monashbioinformaticsplatform.github.io/Functional_Enrichment_Online_Tools//blob/master/02-recap.Rmd">View source <i class="fab fa-gitlab"></i></a></li>
          <li><a id="book-edit" href="https://monashbioinformaticsplatform.github.io/Functional_Enrichment_Online_Tools//edit/master/02-recap.Rmd">Edit this page <i class="fab fa-gitlab"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Functional Enrichment Workshop</strong>" was written by Australian Biocommons, Sydney Informatics Hub (USYD) and Monash Genomics and Bioinformatics Platform (MGBP). It was last built on <span>“October 16, 2025”</span> <br><img src="logos.png" width="300" alt="Logo"><br>.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
