<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 9 Resources | Functional Enrichment Workshop</title>
<meta name="author" content="Australian Biocommons, Sydney Informatics Hub (USYD) and Monash Genomics and Bioinformatics Platform (MGBP)">
<meta name="description" content="Making sense of gene and proteins lists with functional enrichment analysis   The following tools might be useful for downstream functional analysis; this includes some not covered in todays...">
<meta name="generator" content="bookdown 0.40 with bs4_book()">
<meta property="og:title" content="Chapter 9 Resources | Functional Enrichment Workshop">
<meta property="og:type" content="book">
<meta property="og:description" content="Making sense of gene and proteins lists with functional enrichment analysis   The following tools might be useful for downstream functional analysis; this includes some not covered in todays...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 9 Resources | Functional Enrichment Workshop">
<meta name="twitter:description" content="Making sense of gene and proteins lists with functional enrichment analysis   The following tools might be useful for downstream functional analysis; this includes some not covered in todays...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.8.0/transition.js"></script><script src="libs/bs3compat-0.8.0/tabs.js"></script><script src="libs/bs3compat-0.8.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/kePrint-0.0.1/kePrint.js"></script><link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Functional Enrichment Workshop</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> Functional Enrichment Workshop</a></li>
<li><a class="" href="overview.html"><span class="header-section-number">2</span> Overview</a></li>
<li><a class="" href="recap.html"><span class="header-section-number">3</span> Recap</a></li>
<li><a class="" href="enrichment-statistics.html"><span class="header-section-number">4</span> Enrichment Statistics</a></li>
<li><a class="" href="example-analysis.html"><span class="header-section-number">5</span> Example Analysis</a></li>
<li><a class="" href="defining-the-genelist.html"><span class="header-section-number">6</span> Defining the genelist</a></li>
<li><a class="" href="online-tools.html"><span class="header-section-number">7</span> Online Tools</a></li>
<li><a class="" href="reporting-results.html"><span class="header-section-number">8</span> Reporting results</a></li>
<li><a class="active" href="resources.html"><span class="header-section-number">9</span> Resources</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://monashbioinformaticsplatform.github.io/Functional_Enrichment_Online_Tools/">View book source <i class="fab fa-gitlab"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><span class="main-title"><em>Making sense of gene and proteins lists with functional enrichment analysis</em></span>
<div id="resources" class="section level1" number="9">
<h1>
<span class="header-section-number">9</span> Resources<a class="anchor" aria-label="anchor" href="#resources"><i class="fas fa-link"></i></a>
</h1>
<!-- --- -->
<!-- Please find a link to today's presentation slides [here](../../enrichment_analysis_2024.pdf) -->
<!-- ----- -->
<p>The following tools might be useful for downstream functional analysis; this includes some not covered in todays workshop.</p>
<!-- https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/shsy5ydiff.png) 
http://localhost:1313/enrichment_analysis_workshop/img/david.png
-->
<p>Note that most of these tools do more than just enrichment tests, and some include their own databases.</p>
<div id="gprofiler-ggost" class="section level2" number="9.1">
<h2>
<span class="header-section-number">9.1</span> g:Profiler / g:GOSt<a class="anchor" aria-label="anchor" href="#gprofiler-ggost"><i class="fas fa-link"></i></a>
</h2>
<p><a href="https://biit.cs.ut.ee/gprofiler/gost" class="uri">https://biit.cs.ut.ee/gprofiler/gost</a></p>
<p>The <code>g:GOSt</code> ‘functional profiling’ tool of <code>gProfiler</code>calculates functional enrichment for gene lists, identifying enriched terms across various ontologies and pathways.
It features a clean, modern interface and provides a handy summary highlighting which genes contribute to the enrichment.</p>
<!-- ![](https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/gprofiler.png) -->
<div class="figure" style="text-align: centre">
<span style="display:block;" id="fig:unnamed-chunk-42"></span>
<img src="https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/gprofiler.png" alt="The gProfiler front page" width="100%"><p class="caption">
Figure 9.1: The gProfiler front page
</p>
</div>
</div>
<div id="panther" class="section level2" number="9.2">
<h2>
<span class="header-section-number">9.2</span> PANTHER<a class="anchor" aria-label="anchor" href="#panther"><i class="fas fa-link"></i></a>
</h2>
<p><a href="http://www.pantherdb.org/" class="uri">http://www.pantherdb.org/</a></p>
<p>PANTHER performs overrepresentation tests across multiple databases; Gene ontology, reactome, PANTHER pathways and protein classes. Allows more control over the statistical test used and clearly summarises what was actually done.</p>
<!-- ![](https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/panther.png) -->
<div class="figure" style="text-align: centre">
<span style="display:block;" id="fig:unnamed-chunk-43"></span>
<img src="https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/panther.png" alt="PANTHER" width="100%"><p class="caption">
Figure 9.2: PANTHER
</p>
</div>
</div>
<div id="david" class="section level2" number="9.3">
<h2>
<span class="header-section-number">9.3</span> DAVID<a class="anchor" aria-label="anchor" href="#david"><i class="fas fa-link"></i></a>
</h2>
<p><a href="https://david.ncifcrf.gov/" class="uri">https://david.ncifcrf.gov/</a></p>
<p>Via its ‘functional annotation’ tool, DAVID allows you to calculate functional enrichment across a number of databases ; Gene Ontology, KEGG, reactome and others. Reliable, with a slightly clunky interface.</p>
<!-- ![](https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/david.png) -->
<div class="figure" style="text-align: centre">
<span style="display:block;" id="fig:unnamed-chunk-44"></span>
<img src="https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/david.png" alt="DAVID" width="100%"><p class="caption">
Figure 9.3: DAVID
</p>
</div>
</div>
<div id="enrichr" class="section level2" number="9.4">
<h2>
<span class="header-section-number">9.4</span> Enrichr<a class="anchor" aria-label="anchor" href="#enrichr"><i class="fas fa-link"></i></a>
</h2>
<p><a href="https://amp.pharm.mssm.edu/Enrichr/" class="uri">https://amp.pharm.mssm.edu/Enrichr/</a></p>
<p>Enrichr easily calculates enrichment across a wide range of databases. Currently does allow for a background set.</p>
<!-- ![](https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/enrichr.png) -->
<div class="figure" style="text-align: centre">
<span style="display:block;" id="fig:unnamed-chunk-45"></span>
<img src="https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/enrichr.png" alt="Enrichr" width="100%"><p class="caption">
Figure 9.4: Enrichr
</p>
</div>
</div>
<div id="reactome-1" class="section level2" number="9.5">
<h2>
<span class="header-section-number">9.5</span> Reactome<a class="anchor" aria-label="anchor" href="#reactome-1"><i class="fas fa-link"></i></a>
</h2>
<p><a href="https://reactome.org/" class="uri">https://reactome.org/</a></p>
<p>The core of reactome is the reactome pathways and browser. ALthough other tools use the reactome database, the reactome website provides a means to browse enrichment within the pathway browser view.</p>
<!-- ![](https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/reactome.png) -->
<div class="figure" style="text-align: centre">
<span style="display:block;" id="fig:unnamed-chunk-46"></span>
<img src="https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/reactome.png" alt="Reactome" width="100%"><p class="caption">
Figure 9.5: Reactome
</p>
</div>
</div>
<div id="biocyc" class="section level2" number="9.6">
<h2>
<span class="header-section-number">9.6</span> Biocyc<a class="anchor" aria-label="anchor" href="#biocyc"><i class="fas fa-link"></i></a>
</h2>
<p><a href="https://biocyc.org/" class="uri">https://biocyc.org/</a></p>
<p>Biocyc is another suite of tools for enrichment and pathway browsing, which is particularly useful for prokaryotic work. It is licensed, but Monash does have a license.</p>
<!-- ![](https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/biocyc.png) -->
<div class="figure" style="text-align: centre">
<span style="display:block;" id="fig:unnamed-chunk-47"></span>
<img src="https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/biocyc.png" alt="Biocyc" width="100%"><p class="caption">
Figure 9.6: Biocyc
</p>
</div>
</div>
<div id="string" class="section level2" number="9.7">
<h2>
<span class="header-section-number">9.7</span> STRING<a class="anchor" aria-label="anchor" href="#string"><i class="fas fa-link"></i></a>
</h2>
<p><a href="https://string-db.org/" class="uri">https://string-db.org/</a></p>
<p>STRING was originally designed as a convenient tool to explore interactions within a gene list, visualised as an interaction network. While it is best suited for smaller gene lists, it now also includes functionality for performing functional enrichment analysis.</p>
<!-- ![](https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/string.png) -->
<div class="figure" style="text-align: centre">
<span style="display:block;" id="fig:unnamed-chunk-48"></span>
<img src="https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/string.png" alt="STRING" width="100%"><p class="caption">
Figure 9.7: STRING
</p>
</div>
</div>
<div id="gene-ontology" class="section level2" number="9.8">
<h2>
<span class="header-section-number">9.8</span> Gene Ontology<a class="anchor" aria-label="anchor" href="#gene-ontology"><i class="fas fa-link"></i></a>
</h2>
<p><a href="http://geneontology.org/" class="uri">http://geneontology.org/</a></p>
<p>Gene Ontology (GO) terms are the most widely use set of functional annotations, used by many enrichment tools. The gene ontology resource website itself provides several tools for browsing the GO term hierarchy.</p>
<!-- ![](https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/go.png) -->
<div class="figure" style="text-align: centre">
<span style="display:block;" id="fig:unnamed-chunk-49"></span>
<img src="https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/go.png" alt="Gene Ontology" width="100%"><p class="caption">
Figure 9.8: Gene Ontology
</p>
</div>
</div>
<div id="kegg" class="section level2" number="9.9">
<h2>
<span class="header-section-number">9.9</span> KEGG<a class="anchor" aria-label="anchor" href="#kegg"><i class="fas fa-link"></i></a>
</h2>
<p><a href="https://www.genome.jp/kegg/" class="uri">https://www.genome.jp/kegg/</a></p>
<p>A well known curated pathway database. It is used by many other tools but with a caveat - KEGG moved to a subscription model in 2011, and so enrichment tools need to use the last open release from 2011. However up to date KEGG pathways are browsable directly through their website.</p>
<!-- ![](https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/kegg.png) -->
<div class="figure" style="text-align: centre">
<span style="display:block;" id="fig:unnamed-chunk-50"></span>
<img src="https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/kegg.png" alt="KEGG" width="100%"><p class="caption">
Figure 9.9: KEGG
</p>
</div>
</div>
<div id="gsea-and-msigdb" class="section level2" number="9.10">
<h2>
<span class="header-section-number">9.10</span> GSEA and MSigDB<a class="anchor" aria-label="anchor" href="#gsea-and-msigdb"><i class="fas fa-link"></i></a>
</h2>
<p><a href="http://software.broadinstitute.org/gsea/index.jsp" class="uri">http://software.broadinstitute.org/gsea/index.jsp</a></p>
<p>The GSEA Desktop is (one of many) gene set enrichment approaches. It uses of gene rankings across all genes rather than hypogeometric or fishers-exact tests of gene list enrichment. MSigDB (Molecular signatures database) is a suite of annotation databases suitable for GSEA analysis.</p>
<!-- ![](https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/gsea.png) -->
<div class="figure" style="text-align: centre">
<span style="display:block;" id="fig:unnamed-chunk-51"></span>
<img src="https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/gsea.png" alt="MSigDB" width="100%"><p class="caption">
Figure 9.10: MSigDB
</p>
</div>
</div>
<div id="metaboanalyst" class="section level2" number="9.11">
<h2>
<span class="header-section-number">9.11</span> MetaboAnalyst<a class="anchor" aria-label="anchor" href="#metaboanalyst"><i class="fas fa-link"></i></a>
</h2>
<p>MetaboAnalyst is popular among the metabolomics community for statistical, functional and integrative analysis of metabolomics data. It has a feature called <strong>Functional enrichment analysis</strong>, which performs metabolite set enrichment analysis, metabolic pathway analysis, and pathway activity prediction from MS peaks.</p>
<!-- ![](https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/metaboanalyst.png) -->
<div class="figure" style="text-align: centre">
<span style="display:block;" id="fig:unnamed-chunk-52"></span>
<img src="https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/metaboanalyst.png" alt="MetaboAnalyst" width="100%"><p class="caption">
Figure 9.11: MetaboAnalyst
</p>
</div>
</div>
<div id="cytoscape" class="section level2" number="9.12">
<h2>
<span class="header-section-number">9.12</span> Cytoscape<a class="anchor" aria-label="anchor" href="#cytoscape"><i class="fas fa-link"></i></a>
</h2>
<p><a href="https://cytoscape.org/" class="uri">https://cytoscape.org/</a></p>
<p>Cytoscape is a desktop-based biological network analysis / visualisation tool, rather than a functional enrichment tool (although plugins can change that). It is mentioned here because it is often useful as a next step when you need to create custom figures showing the interactions of an interesting biological pathway.</p>
<!-- ![](https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/cytoscape.png) -->
<div class="figure" style="text-align: centre">
<span style="display:block;" id="fig:unnamed-chunk-53"></span>
<img src="https://monashbioinformaticsplatform.github.io/enrichment_analysis_workshop/img/cytoscape.png" alt="Cytoscape" width="100%"><p class="caption">
Figure 9.12: Cytoscape
</p>
</div>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="reporting-results.html"><span class="header-section-number">8</span> Reporting results</a></div>
<div class="empty"></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#resources"><span class="header-section-number">9</span> Resources</a></li>
<li><a class="nav-link" href="#gprofiler-ggost"><span class="header-section-number">9.1</span> g:Profiler / g:GOSt</a></li>
<li><a class="nav-link" href="#panther"><span class="header-section-number">9.2</span> PANTHER</a></li>
<li><a class="nav-link" href="#david"><span class="header-section-number">9.3</span> DAVID</a></li>
<li><a class="nav-link" href="#enrichr"><span class="header-section-number">9.4</span> Enrichr</a></li>
<li><a class="nav-link" href="#reactome-1"><span class="header-section-number">9.5</span> Reactome</a></li>
<li><a class="nav-link" href="#biocyc"><span class="header-section-number">9.6</span> Biocyc</a></li>
<li><a class="nav-link" href="#string"><span class="header-section-number">9.7</span> STRING</a></li>
<li><a class="nav-link" href="#gene-ontology"><span class="header-section-number">9.8</span> Gene Ontology</a></li>
<li><a class="nav-link" href="#kegg"><span class="header-section-number">9.9</span> KEGG</a></li>
<li><a class="nav-link" href="#gsea-and-msigdb"><span class="header-section-number">9.10</span> GSEA and MSigDB</a></li>
<li><a class="nav-link" href="#metaboanalyst"><span class="header-section-number">9.11</span> MetaboAnalyst</a></li>
<li><a class="nav-link" href="#cytoscape"><span class="header-section-number">9.12</span> Cytoscape</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://monashbioinformaticsplatform.github.io/Functional_Enrichment_Online_Tools//blob/master/08-resources.Rmd">View source <i class="fab fa-gitlab"></i></a></li>
          <li><a id="book-edit" href="https://monashbioinformaticsplatform.github.io/Functional_Enrichment_Online_Tools//edit/master/08-resources.Rmd">Edit this page <i class="fab fa-gitlab"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Functional Enrichment Workshop</strong>" was written by Australian Biocommons, Sydney Informatics Hub (USYD) and Monash Genomics and Bioinformatics Platform (MGBP). It was last built on <span>“October 14, 2025”</span> <br><img src="logos.png" width="300" alt="Logo"><br>.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
