<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 18 Workshop wrap-up | Functional Enrichment Workshop</title>
<meta name="author" content="Australian Biocommons, Sydney Informatics Hub (USYD) and Monash Genomics and Bioinformatics Platform (MGBP)">
<meta name="description" content="Making sense of gene and proteins lists with functional enrichment analysis  Over the last two days, as well as the webinar in October, we‚Äôve explored the statistical background, key...">
<meta name="generator" content="bookdown 0.40 with bs4_book()">
<meta property="og:title" content="Chapter 18 Workshop wrap-up | Functional Enrichment Workshop">
<meta property="og:type" content="book">
<meta property="og:description" content="Making sense of gene and proteins lists with functional enrichment analysis  Over the last two days, as well as the webinar in October, we‚Äôve explored the statistical background, key...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 18 Workshop wrap-up | Functional Enrichment Workshop">
<meta name="twitter:description" content="Making sense of gene and proteins lists with functional enrichment analysis  Over the last two days, as well as the webinar in October, we‚Äôve explored the statistical background, key...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.8.0/transition.js"></script><script src="libs/bs3compat-0.8.0/tabs.js"></script><script src="libs/bs3compat-0.8.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/kePrint-0.0.1/kePrint.js"></script><link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Functional Enrichment Workshop</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> Functional Enrichment Workshop</a></li>
<li class="book-part">Day 1</li>
<li><a class="" href="overview.html"><span class="header-section-number">3</span> Overview</a></li>
<li><a class="" href="recap.html"><span class="header-section-number">4</span> Recap</a></li>
<li><a class="" href="enrichment-statistics.html"><span class="header-section-number">5</span> Enrichment Statistics</a></li>
<li><a class="" href="example-analysis.html"><span class="header-section-number">6</span> Example Analysis</a></li>
<li><a class="" href="defining-the-genelist.html"><span class="header-section-number">7</span> Defining the genelist</a></li>
<li><a class="" href="online-tools.html"><span class="header-section-number">8</span> Online Tools</a></li>
<li><a class="" href="reporting-results.html"><span class="header-section-number">9</span> Reporting results</a></li>
<li><a class="" href="resources.html"><span class="header-section-number">10</span> Resources</a></li>
<li class="book-part">Day 2</li>
<li><a class="" href="introduction.html"><span class="header-section-number">12</span> Introduction</a></li>
<li><a class="" href="r-environment-set-up.html"><span class="header-section-number">13</span> R environment set up</a></li>
<li><a class="" href="gprofiler2.html"><span class="header-section-number">14</span> gprofiler2</a></li>
<li><a class="" href="clusterprofiler-and-enrichplot.html"><span class="header-section-number">15</span> clusterProfiler and enrichplot</a></li>
<li><a class="" href="webgestaltr.html"><span class="header-section-number">16</span> WebGestaltR</a></li>
<li><a class="" href="novel-species-fea.html"><span class="header-section-number">17</span> Novel species FEA</a></li>
<li><a class="active" href="workshop-wrap-up.html"><span class="header-section-number">18</span> Workshop wrap-up</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://monashbioinformaticsplatform.github.io/Functional_Enrichment_Online_Tools/">View book source <i class="fab fa-gitlab"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><span class="main-title"><em>Making sense of gene and proteins lists with functional enrichment analysis</em></span>
<div id="workshop-wrap-up" class="section level1" number="18">
<h1>
<span class="header-section-number">18</span> Workshop wrap-up<a class="anchor" aria-label="anchor" href="#workshop-wrap-up"><i class="fas fa-link"></i></a>
</h1>
<p>Over the last two days, as well as the webinar in October, we‚Äôve explored the statistical background, key considerations, and practical implementation of functional enrichment analysis, with hands-on experience with multiple web-based and R tools.</p>
<div id="summary-of-key-messages" class="section level2" number="18.1">
<h2>
<span class="header-section-number">18.1</span> Summary of key messages<a class="anchor" aria-label="anchor" href="#summary-of-key-messages"><i class="fas fa-link"></i></a>
</h2>
<p><strong>ORA and GSEA are different statistical analyses, and their inputs differ</strong></p>
<p>GSEA: Kolmogorov-Smirnov test, requires a ranked yet unfiltered gene list
ORA: Hypergeometric or Fisher‚Äôs Exact test, requires a filtered unranked gene list and experimental background gene list</p>
<p><strong>Always correct for multiple testing</strong></p>
<p>Never use unadjusted P values, as this will introduce many false positives. Different tools offer different multiple testing correction such as FDR or the more stringent BH. Always report your chosen method and the significance threshold applied to terms.</p>
<p><strong>Different analysis methods will return different results</strong></p>
<p>This is expected, due to underlying differences in database, algorithm, P value methods etc. As long as your methods are robust, sensible and reproducible, you can have confidence that your methods will stand up to scrutiny under peer review.</p>
<p><strong>Ensure reproducibility</strong></p>
<p>Lack of reproducibility through under-reporting methods is a common issue in this field (see Wijesooriya et al, linked below). Ensure to record all methodological details while you are working, including all the parameters and arguments applied, how the gene lists were generated, versions of databases and tools etc. If using R, specify a seed for constant random number generation in GSEA.</p>
<p><strong>Interpret your results in their biological context</strong></p>
<p>Functional categories are often broad and redundant. Use the FEA results as a guide, not the end point. Use visualisations and explore term redundancy methods to help focus results. Validate through aditional means according to the nature of your experiment, with the gold standard being wet-lab rather than <em>in silico</em> validation methods. Keep in mind the limitations of the input data when working with novel species with uncurated resources.</p>
<p><strong>There are many databases and tool choices available</strong></p>
<p>Suitability to your experiment depends on many factors, including:</p>
<ul>
<li>Your species, and what tools support it</li>
<li>What databases and gene sets are relevant to your experiment, from the general (eg GO) to the specific (eg cancer pathways)</li>
<li>Any privacy restrictions imposed on your data</li>
<li>What is your skill level in R or desire to implement R code</li>
<li>How much flexibility you want or require with visualisation</li>
</ul>
</div>
<div id="relevant-papers-for-further-reading" class="section level2" number="18.2">
<h2>
<span class="header-section-number">18.2</span> Relevant papers for further reading<a class="anchor" aria-label="anchor" href="#relevant-papers-for-further-reading"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li><p><a href="https://www.sciencedirect.com/science/article/abs/pii/S0168952523000185">Interpreting omics data with pathway enrichment analysis</a></p></li>
<li><p><a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1009935">Urgent need for consistent standards in functional enrichment analysis</a></p></li>
<li><p><a href="https://academic.oup.com/bib/article/23/5/bbac355/6695266">Systematic assessment of pathway databases, based on a diverse collection of user-submitted experiments</a></p></li>
<li><p><a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-015-0761-7">Multiple sources of bias confound functional enrichment analysis of global -omics data</a></p></li>
<li><p><a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC6883970/">The Impact of Pathway Database Choice on Statistical Enrichment Analysis and Predictive Modeling</a></p></li>
<li><p><a href="https://academic.oup.com/bioinformaticsadvances/article/4/1/vbae159/7829164?login=false">Two subtle problems with overrepresentation analysis</a></p></li>
<li><p><a href="https://biodatamining.biomedcentral.com/articles/10.1186/s13040-018-0166-8">Gene set analysis methods: A systematic comparison</a></p></li>
<li><p><a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0079217">A Comparison of Gene Set Analysis Methods in Terms of Sensitivity, Prioritization and Specificity</a></p></li>
<li><p><a href="https://link.springer.com/article/10.1007/s10571-016-0403-y">Transcriptomic Profiling Discloses Molecular and Cellular Events Related to Neuronal Differentiation in SH-SY5Y Neuroblastoma Cells</a></p></li>
<li><p><a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1002375">Ten Years of Pathway Analysis: Current Approaches and Outstanding Challenges</a></p></li>
<li><p><a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0191154">A critical comparison of topology-based pathway analysis methods</a></p></li>
<li><p><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-017-1674-0">Ranking metrics in Gene Set Enrichment Analysis: Do they matter?</a></p></li>
<li><p><a href="https://www.frontiersin.org/journals/physiology/articles/10.3389/fphys.2013.00278/full">Methods and approaches in the topology-based analysis of biological pathways</a></p></li>
<li><p><a href="https://academic.oup.com/bib/article/15/4/504/407653">Gene set analysis methods: Statistical models and methodological differences</a></p></li>
<li><p><a href="https://pubmed.ncbi.nlm.nih.gov/31684881/">A Comparative Study of Topology-based Pathway Enrichment Analysis Methods</a></p></li>
<li><p><a href="https://academic.oup.com/bib/article/22/1/545/5722384">Toward a Gold Standard for Benchmarking Gene Set Enrichment Analysis</a></p></li>
<li><p><a href="https://www.pnas.org/content/102/43/15545">Gene Set Enrichment Analysis: A Knowledge-Based Approach for Interpreting Genome-Wide Expression Profiles</a></p></li>
<li><p><a href="https://arxiv.org/abs/2308.15171">From RNA sequencing measurements to the final results: A practical guide to navigating the choices and uncertainties of gene set analysis</a></p></li>
</ul>
</div>
<div id="audience-poll" class="section level2" number="18.3">
<h2>
<span class="header-section-number">18.3</span> Audience poll<a class="anchor" aria-label="anchor" href="#audience-poll"><i class="fas fa-link"></i></a>
</h2>
<p>There is of course no correct answer here, we are just interested to hear your thoughts!</p>
<p><strong><span style="color: green;">‚ÄúIf you were to run FEA on your own data tomorrow, would you choose web or R tools?‚Äù</span></strong> ü§î</p>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="novel-species-fea.html"><span class="header-section-number">17</span> Novel species FEA</a></div>
<div class="empty"></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#workshop-wrap-up"><span class="header-section-number">18</span> Workshop wrap-up</a></li>
<li><a class="nav-link" href="#summary-of-key-messages"><span class="header-section-number">18.1</span> Summary of key messages</a></li>
<li><a class="nav-link" href="#relevant-papers-for-further-reading"><span class="header-section-number">18.2</span> Relevant papers for further reading</a></li>
<li><a class="nav-link" href="#audience-poll"><span class="header-section-number">18.3</span> Audience poll</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://monashbioinformaticsplatform.github.io/Functional_Enrichment_Online_Tools//blob/master/15-workshop-wrapup.Rmd">View source <i class="fab fa-gitlab"></i></a></li>
          <li><a id="book-edit" href="https://monashbioinformaticsplatform.github.io/Functional_Enrichment_Online_Tools//edit/master/15-workshop-wrapup.Rmd">Edit this page <i class="fab fa-gitlab"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Functional Enrichment Workshop</strong>" was written by Australian Biocommons, Sydney Informatics Hub (USYD) and Monash Genomics and Bioinformatics Platform (MGBP). It was last built on <span>‚ÄúOctober 14, 2025‚Äù</span> <br><img src="logos.png" width="300" alt="Logo"><br>.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
